{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": [
        "6guqvAjSIiKp",
        "3HYCRktUMx45"
      ]
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Подготовка"
      ],
      "metadata": {
        "id": "2MdjyFLFJ45m"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hwem9tcsJRtz",
        "outputId": "cdba8b3c-9904-432d-e81e-ac86211ec36e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting biopython\n",
            "  Downloading biopython-1.81-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.1/3.1 MB\u001b[0m \u001b[31m24.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from biopython) (1.22.4)\n",
            "Installing collected packages: biopython\n",
            "Successfully installed biopython-1.81\n"
          ]
        }
      ],
      "source": [
        "!pip install biopython"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
        "!tar -zxvf bedtools-2.30.0.tar.gz\n",
        "!cd bedtools2; make\n",
        "!cp /content/bedtools2/bin/* /usr/local/bin/"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aM0VV3s5K4AJ",
        "outputId": "f2361926-2978-45dc-8214-965cdff08ebd"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-05-16 22:38:21--  http://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Resolving github.com (github.com)... 192.30.255.112\n",
            "Connecting to github.com (github.com)|192.30.255.112|:80... connected.\n",
            "HTTP request sent, awaiting response... 301 Moved Permanently\n",
            "Location: https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz [following]\n",
            "--2023-05-16 22:38:21--  https://github.com/arq5x/bedtools2/releases/download/v2.30.0/bedtools-2.30.0.tar.gz\n",
            "Connecting to github.com (github.com)|192.30.255.112|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20230516%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230516T223821Z&X-Amz-Expires=300&X-Amz-Signature=97e1b60cd20b79c5cc12cb68b9764bc0fffde0e27ef509fa3e8c402f3cc27ffa&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2023-05-16 22:38:21--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/15059334/aecc9080-5d79-11eb-847e-0ff72e8f556a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20230516%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230516T223821Z&X-Amz-Expires=300&X-Amz-Signature=97e1b60cd20b79c5cc12cb68b9764bc0fffde0e27ef509fa3e8c402f3cc27ffa&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=15059334&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.30.0.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 20648577 (20M) [application/octet-stream]\n",
            "Saving to: ‘bedtools-2.30.0.tar.gz’\n",
            "\n",
            "bedtools-2.30.0.tar 100%[===================>]  19.69M  62.7MB/s    in 0.3s    \n",
            "\n",
            "2023-05-16 22:38:22 (62.7 MB/s) - ‘bedtools-2.30.0.tar.gz’ saved [20648577/20648577]\n",
            "\n",
            "bedtools2/\n",
            "bedtools2/obj/\n",
            "bedtools2/tutorial/\n",
            "bedtools2/LICENSE\n",
            "bedtools2/test/\n",
            "bedtools2/bin/\n",
            "bedtools2/Makefile\n",
            "bedtools2/docs/\n",
            "bedtools2/README.md\n",
            "bedtools2/genomes/\n",
            "bedtools2/.gitignore\n",
            "bedtools2/scripts/\n",
            "bedtools2/.github/\n",
            "bedtools2/data/\n",
            "bedtools2/src/\n",
            "bedtools2/src/spacingFile/\n",
            "bedtools2/src/sortBed/\n",
            "bedtools2/src/split/\n",
            "bedtools2/src/bedToBam/\n",
            "bedtools2/src/complementFile/\n",
            "bedtools2/src/expand/\n",
            "bedtools2/src/fisher/\n",
            "bedtools2/src/linksBed/\n",
            "bedtools2/src/pairToPair/\n",
            "bedtools2/src/bedtools.cpp\n",
            "bedtools2/src/mapFile/\n",
            "bedtools2/src/slopBed/\n",
            "bedtools2/src/annotateBed/\n",
            "bedtools2/src/randomBed/\n",
            "bedtools2/src/getOverlap/\n",
            "bedtools2/src/reldist/\n",
            "bedtools2/src/nucBed/\n",
            "bedtools2/src/bedToIgv/\n",
            "bedtools2/src/regressTest/\n",
            "bedtools2/src/maskFastaFromBed/\n",
            "bedtools2/src/genomeCoverageBed/\n",
            "bedtools2/src/utils/\n",
            "bedtools2/src/windowBed/\n",
            "bedtools2/src/jaccard/\n",
            "bedtools2/src/groupBy/\n",
            "bedtools2/src/bamToFastq/\n",
            "bedtools2/src/htsutil.cpp\n",
            "bedtools2/src/fastaFromBed/\n",
            "bedtools2/src/nekSandbox1/\n",
            "bedtools2/src/unionBedGraphs/\n",
            "bedtools2/src/bed12ToBed6/\n",
            "bedtools2/src/subtractFile/\n",
            "bedtools2/src/bedpeToBam/\n",
            "bedtools2/src/windowMaker/\n",
            "bedtools2/src/coverageFile/\n",
            "bedtools2/src/summaryFile/\n",
            "bedtools2/src/multiIntersectBed/\n",
            "bedtools2/src/intersectFile/\n",
            "bedtools2/src/closestFile/\n",
            "bedtools2/src/shuffleBed/\n",
            "bedtools2/src/flankBed/\n",
            "bedtools2/src/tagBam/\n",
            "bedtools2/src/multiBamCov/\n",
            "bedtools2/src/sampleFile/\n",
            "bedtools2/src/clusterBed/\n",
            "bedtools2/src/shiftBed/\n",
            "bedtools2/src/mergeFile/\n",
            "bedtools2/src/bamToBed/\n",
            "bedtools2/src/pairToBed/\n",
            "bedtools2/src/pairToBed/pairToBed.h\n",
            "bedtools2/src/pairToBed/pairToBedMain.cpp\n",
            "bedtools2/src/pairToBed/Makefile.frag\n",
            "bedtools2/src/pairToBed/pairToBed.cpp\n",
            "bedtools2/src/bamToBed/bamToBed.cpp\n",
            "bedtools2/src/bamToBed/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeFile.cpp\n",
            "bedtools2/src/mergeFile/Makefile.frag\n",
            "bedtools2/src/mergeFile/mergeHelp.cpp\n",
            "bedtools2/src/mergeFile/mergeFile.h\n",
            "bedtools2/src/shiftBed/shiftBed.cpp\n",
            "bedtools2/src/shiftBed/shiftBedMain.cpp\n",
            "bedtools2/src/shiftBed/Makefile.frag\n",
            "bedtools2/src/shiftBed/shiftBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.h\n",
            "bedtools2/src/clusterBed/clusterBed.cpp\n",
            "bedtools2/src/clusterBed/clusterMain.cpp\n",
            "bedtools2/src/clusterBed/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleFile.cpp\n",
            "bedtools2/src/sampleFile/sampleFile.h\n",
            "bedtools2/src/sampleFile/Makefile.frag\n",
            "bedtools2/src/sampleFile/sampleHelp.cpp\n",
            "bedtools2/src/multiBamCov/multiBamCov.cpp\n",
            "bedtools2/src/multiBamCov/Makefile.frag\n",
            "bedtools2/src/multiBamCov/multiBamCov.h\n",
            "bedtools2/src/multiBamCov/multiBamCovMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.cpp\n",
            "bedtools2/src/tagBam/tagBamMain.cpp\n",
            "bedtools2/src/tagBam/tagBam.h\n",
            "bedtools2/src/tagBam/Makefile.frag\n",
            "bedtools2/src/flankBed/flankBed.cpp\n",
            "bedtools2/src/flankBed/flankBedMain.cpp\n",
            "bedtools2/src/flankBed/test.fa.fai\n",
            "bedtools2/src/flankBed/test.gff\n",
            "bedtools2/src/flankBed/Makefile.frag\n",
            "bedtools2/src/flankBed/test.fa\n",
            "bedtools2/src/flankBed/flankBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.h\n",
            "bedtools2/src/shuffleBed/shuffleBed.cpp\n",
            "bedtools2/src/shuffleBed/Makefile.frag\n",
            "bedtools2/src/shuffleBed/shuffleBedMain.cpp\n",
            "bedtools2/src/closestFile/closestHelp.cpp\n",
            "bedtools2/src/closestFile/closestFile.h\n",
            "bedtools2/src/closestFile/closestFile.cpp\n",
            "bedtools2/src/closestFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectHelp.cpp\n",
            "bedtools2/src/intersectFile/Makefile.frag\n",
            "bedtools2/src/intersectFile/intersectFile.cpp\n",
            "bedtools2/src/intersectFile/intersectFile.h\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.cpp\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBed.h\n",
            "bedtools2/src/multiIntersectBed/Makefile.frag\n",
            "bedtools2/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "bedtools2/src/summaryFile/summaryHelp.cpp\n",
            "bedtools2/src/summaryFile/Makefile.frag\n",
            "bedtools2/src/summaryFile/summaryFile.h\n",
            "bedtools2/src/summaryFile/summaryFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.cpp\n",
            "bedtools2/src/coverageFile/coverageFile.h\n",
            "bedtools2/src/coverageFile/Makefile.frag\n",
            "bedtools2/src/coverageFile/coverageHelp.cpp\n",
            "bedtools2/src/windowMaker/windowMakerMain.cpp\n",
            "bedtools2/src/windowMaker/windowMaker.h\n",
            "bedtools2/src/windowMaker/windowMaker.cpp\n",
            "bedtools2/src/windowMaker/Makefile.frag\n",
            "bedtools2/src/bedpeToBam/notes.txt\n",
            "bedtools2/src/bedpeToBam/bedpeToBam.cpp\n",
            "bedtools2/src/bedpeToBam/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractHelp.cpp\n",
            "bedtools2/src/subtractFile/subtractFile.h\n",
            "bedtools2/src/subtractFile/Makefile.frag\n",
            "bedtools2/src/subtractFile/subtractFile.cpp\n",
            "bedtools2/src/bed12ToBed6/bed12ToBed6.cpp\n",
            "bedtools2/src/bed12ToBed6/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.cpp\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "bedtools2/src/unionBedGraphs/Makefile.frag\n",
            "bedtools2/src/unionBedGraphs/unionBedGraphs.h\n",
            "bedtools2/src/nekSandbox1/FormatGuess.h\n",
            "bedtools2/src/nekSandbox1/Makefile\n",
            "bedtools2/src/nekSandbox1/nekSandboxMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBedMain.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.cpp\n",
            "bedtools2/src/fastaFromBed/fastaFromBed.h\n",
            "bedtools2/src/fastaFromBed/Makefile.frag\n",
            "bedtools2/src/bamToFastq/bamToFastq.h\n",
            "bedtools2/src/bamToFastq/bamToFastq.cpp\n",
            "bedtools2/src/bamToFastq/bamToFastqMain.cpp\n",
            "bedtools2/src/bamToFastq/Makefile.frag\n",
            "bedtools2/src/groupBy/groupBy.cpp\n",
            "bedtools2/src/groupBy/groupBy.h\n",
            "bedtools2/src/groupBy/Makefile.frag\n",
            "bedtools2/src/groupBy/groupByHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.cpp\n",
            "bedtools2/src/jaccard/Makefile.frag\n",
            "bedtools2/src/jaccard/jaccardHelp.cpp\n",
            "bedtools2/src/jaccard/jaccard.h\n",
            "bedtools2/src/windowBed/windowBed.cpp\n",
            "bedtools2/src/windowBed/windowBed.h\n",
            "bedtools2/src/windowBed/windowMain.cpp\n",
            "bedtools2/src/windowBed/Makefile.frag\n",
            "bedtools2/src/utils/tabFile/\n",
            "bedtools2/src/utils/htslib/\n",
            "bedtools2/src/utils/NewChromsweep/\n",
            "bedtools2/src/utils/RecordOutputMgr/\n",
            "bedtools2/src/utils/bedFile/\n",
            "bedtools2/src/utils/Contexts/\n",
            "bedtools2/src/utils/BamTools/\n",
            "bedtools2/src/utils/BlockedIntervals/\n",
            "bedtools2/src/utils/general/\n",
            "bedtools2/src/utils/driver/\n",
            "bedtools2/src/utils/Point/\n",
            "bedtools2/src/utils/bedGraphFile/\n",
            "bedtools2/src/utils/sequenceUtilities/\n",
            "bedtools2/src/utils/Fasta/\n",
            "bedtools2/src/utils/gzstream/\n",
            "bedtools2/src/utils/fileType/\n",
            "bedtools2/src/utils/KeyListOps/\n",
            "bedtools2/src/utils/BamTools-Ancillary/\n",
            "bedtools2/src/utils/GenomeFile/\n",
            "bedtools2/src/utils/version/\n",
            "bedtools2/src/utils/ToolBase/\n",
            "bedtools2/src/utils/lineFileUtilities/\n",
            "bedtools2/src/utils/bedFilePE/\n",
            "bedtools2/src/utils/chromsweep/\n",
            "bedtools2/src/utils/FileRecordTools/\n",
            "bedtools2/src/utils/BinTree/\n",
            "bedtools2/src/utils/VectorOps/\n",
            "bedtools2/src/utils/stringUtilities/\n",
            "bedtools2/src/utils/stringUtilities/stringUtilities.h\n",
            "bedtools2/src/utils/VectorOps/VectorOps.cpp\n",
            "bedtools2/src/utils/VectorOps/Makefile.frag\n",
            "bedtools2/src/utils/VectorOps/VectorOps.h\n",
            "bedtools2/src/utils/BinTree/BinTree.cpp\n",
            "bedtools2/src/utils/BinTree/BinTree.h\n",
            "bedtools2/src/utils/BinTree/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/\n",
            "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
            "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Makefile.frag\n",
            "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Record.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BamRecord.h\n",
            "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
            "bedtools2/src/utils/chromsweep/chromsweep.cpp\n",
            "bedtools2/src/utils/chromsweep/Makefile.frag\n",
            "bedtools2/src/utils/chromsweep/chromsweep.h\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.cpp\n",
            "bedtools2/src/utils/bedFilePE/Makefile.frag\n",
            "bedtools2/src/utils/bedFilePE/bedFilePE.h\n",
            "bedtools2/src/utils/lineFileUtilities/lineFileUtilities.h\n",
            "bedtools2/src/utils/ToolBase/ToolBase.cpp\n",
            "bedtools2/src/utils/ToolBase/Makefile.frag\n",
            "bedtools2/src/utils/ToolBase/ToolBase.h\n",
            "bedtools2/src/utils/version/version.h\n",
            "bedtools2/src/utils/version/version.cpp\n",
            "bedtools2/src/utils/version/version_release.txt\n",
            "bedtools2/src/utils/version/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/NewGenomeFile.h\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.cpp\n",
            "bedtools2/src/utils/GenomeFile/Makefile.frag\n",
            "bedtools2/src/utils/GenomeFile/GenomeFile.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "bedtools2/src/utils/BamTools-Ancillary/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.h\n",
            "bedtools2/src/utils/KeyListOps/KeyListOps.cpp\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "bedtools2/src/utils/KeyListOps/Makefile.frag\n",
            "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.h\n",
            "bedtools2/src/utils/fileType/fileType.cpp\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.h\n",
            "bedtools2/src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "bedtools2/src/utils/fileType/Makefile.frag\n",
            "bedtools2/src/utils/fileType/fileType.h\n",
            "bedtools2/src/utils/gzstream/gzstream.C\n",
            "bedtools2/src/utils/gzstream/README\n",
            "bedtools2/src/utils/gzstream/COPYING.LIB\n",
            "bedtools2/src/utils/gzstream/Makefile.frag\n",
            "bedtools2/src/utils/gzstream/version.txt\n",
            "bedtools2/src/utils/gzstream/gzstream.h\n",
            "bedtools2/src/utils/Fasta/Fasta.h\n",
            "bedtools2/src/utils/Fasta/split.cpp\n",
            "bedtools2/src/utils/Fasta/Fasta.cpp\n",
            "bedtools2/src/utils/Fasta/LargeFileSupport.h\n",
            "bedtools2/src/utils/Fasta/Makefile.frag\n",
            "bedtools2/src/utils/Fasta/split.h\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.h\n",
            "bedtools2/src/utils/sequenceUtilities/Makefile.frag\n",
            "bedtools2/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.h\n",
            "bedtools2/src/utils/bedGraphFile/Makefile.frag\n",
            "bedtools2/src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "bedtools2/src/utils/Point/Point.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.h\n",
            "bedtools2/src/utils/driver/BedtoolsDriver.cpp\n",
            "bedtools2/src/utils/driver/Makefile.frag\n",
            "bedtools2/src/utils/general/QuickString.cpp\n",
            "bedtools2/src/utils/general/ParseTools.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.h\n",
            "bedtools2/src/utils/general/FreeList.h\n",
            "bedtools2/src/utils/general/CompressionTools.cpp\n",
            "bedtools2/src/utils/general/Random.cpp\n",
            "bedtools2/src/utils/general/BedtoolsTypes.h\n",
            "bedtools2/src/utils/general/ErrorMsg.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.h\n",
            "bedtools2/src/utils/general/Tokenizer.h\n",
            "bedtools2/src/utils/general/ParseTools.h\n",
            "bedtools2/src/utils/general/QuickString.h\n",
            "bedtools2/src/utils/general/Tokenizer.cpp\n",
            "bedtools2/src/utils/general/ErrorMsg.cpp\n",
            "bedtools2/src/utils/general/CommonHelp.cpp\n",
            "bedtools2/src/utils/general/Makefile.frag\n",
            "bedtools2/src/utils/general/BTlist.h\n",
            "bedtools2/src/utils/general/CompressionTools.h\n",
            "bedtools2/src/utils/general/PushBackStreamBuf.cpp\n",
            "bedtools2/src/utils/general/InflateStreamBuf.h\n",
            "bedtools2/src/utils/general/Random.h\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.h\n",
            "bedtools2/src/utils/BlockedIntervals/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/mapping/\n",
            "bedtools2/src/utils/BamTools/include/\n",
            "bedtools2/src/utils/BamTools/.gitignore\n",
            "bedtools2/src/utils/BamTools/Makefile.frag\n",
            "bedtools2/src/utils/BamTools/include/BamAncillary.h\n",
            "bedtools2/src/utils/BamTools/include/SamHeader.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamReader.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/\n",
            "bedtools2/src/utils/BamTools/include/BamWriter.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.mapping.hpp\n",
            "bedtools2/src/utils/BamTools/include/BamAlignment.hpp\n",
            "bedtools2/src/utils/BamTools/include/api/BamWriter.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamMultiReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/SamConstants.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAlignment.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamReader.h\n",
            "bedtools2/src/utils/BamTools/include/api/BamAux.h\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.map\n",
            "bedtools2/src/utils/BamTools/mapping/BamAlignment.py\n",
            "bedtools2/src/utils/Contexts/ContextBase.h\n",
            "bedtools2/src/utils/Contexts/ContextClosest.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.h\n",
            "bedtools2/src/utils/Contexts/ContextIntersect.cpp\n",
            "bedtools2/src/utils/Contexts/ContextFisher.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMerge.h\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.h\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.cpp\n",
            "bedtools2/src/utils/Contexts/ContextBase.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSummary.h\n",
            "bedtools2/src/utils/Contexts/ContextFisher.h\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.h\n",
            "bedtools2/src/utils/Contexts/ContextSubtract.cpp\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.h\n",
            "bedtools2/src/utils/Contexts/ContextComplement.cpp\n",
            "bedtools2/src/utils/Contexts/ContextCoverage.h\n",
            "bedtools2/src/utils/Contexts/ContextSample.cpp\n",
            "bedtools2/src/utils/Contexts/ContextClosest.cpp\n",
            "bedtools2/src/utils/Contexts/Makefile.frag\n",
            "bedtools2/src/utils/Contexts/ContextJaccard.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSample.h\n",
            "bedtools2/src/utils/Contexts/ContextSummary.cpp\n",
            "bedtools2/src/utils/Contexts/ContextGroupBy.cpp\n",
            "bedtools2/src/utils/Contexts/ContextMap.cpp\n",
            "bedtools2/src/utils/Contexts/ContextSpacing.h\n",
            "bedtools2/src/utils/Contexts/ContextMerge.cpp\n",
            "bedtools2/src/utils/bedFile/bedFile.h\n",
            "bedtools2/src/utils/bedFile/Makefile.frag\n",
            "bedtools2/src/utils/bedFile/bedFile.cpp\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
            "bedtools2/src/utils/RecordOutputMgr/Makefile.frag\n",
            "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.h\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.h\n",
            "bedtools2/src/utils/NewChromsweep/Makefile.frag\n",
            "bedtools2/src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "bedtools2/src/utils/NewChromsweep/CloseSweep.cpp\n",
            "bedtools2/src/utils/htslib/htslib.mk\n",
            "bedtools2/src/utils/htslib/cram/\n",
            "bedtools2/src/utils/htslib/htslib/\n",
            "bedtools2/src/utils/htslib/faidx.c\n",
            "bedtools2/src/utils/htslib/vcfutils.c\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.h\n",
            "bedtools2/src/utils/htslib/faidx.5\n",
            "bedtools2/src/utils/htslib/regidx.c\n",
            "bedtools2/src/utils/htslib/htslib.pc.in\n",
            "bedtools2/src/utils/htslib/configure.ac\n",
            "bedtools2/src/utils/htslib/INSTALL\n",
            "bedtools2/src/utils/htslib/hts_os.c\n",
            "bedtools2/src/utils/htslib/tbx.c\n",
            "bedtools2/src/utils/htslib/probaln.c\n",
            "bedtools2/src/utils/htslib/LICENSE\n",
            "bedtools2/src/utils/htslib/kfunc.c\n",
            "bedtools2/src/utils/htslib/hfile_gcs.c\n",
            "bedtools2/src/utils/htslib/errmod.c\n",
            "bedtools2/src/utils/htslib/multipart.c\n",
            "bedtools2/src/utils/htslib/hfile_internal.h\n",
            "bedtools2/src/utils/htslib/Makefile\n",
            "bedtools2/src/utils/htslib/htsfile.c\n",
            "bedtools2/src/utils/htslib/realn.c\n",
            "bedtools2/src/utils/htslib/bgzip.c\n",
            "bedtools2/src/utils/htslib/hfile_net.c\n",
            "bedtools2/src/utils/htslib/tabix.1\n",
            "bedtools2/src/utils/htslib/textutils.c\n",
            "bedtools2/src/utils/htslib/hts.c\n",
            "bedtools2/src/utils/htslib/README\n",
            "bedtools2/src/utils/htslib/vcf_sweep.c\n",
            "bedtools2/src/utils/htslib/README.md\n",
            "bedtools2/src/utils/htslib/sam.5\n",
            "bedtools2/src/utils/htslib/bcf_sr_sort.c\n",
            "bedtools2/src/utils/htslib/synced_bcf_reader.c\n",
            "bedtools2/src/utils/htslib/htslib_vars.mk\n",
            "bedtools2/src/utils/htslib/sam.c\n",
            "bedtools2/src/utils/htslib/NEWS\n",
            "bedtools2/src/utils/htslib/.gitignore\n",
            "bedtools2/src/utils/htslib/vcf.5\n",
            "bedtools2/src/utils/htslib/bgzf.c\n",
            "bedtools2/src/utils/htslib/.gitattributes\n",
            "bedtools2/src/utils/htslib/hts_internal.h\n",
            "bedtools2/src/utils/htslib/vcf.c\n",
            "bedtools2/src/utils/htslib/m4/\n",
            "bedtools2/src/utils/htslib/plugin.c\n",
            "bedtools2/src/utils/htslib/version.sh\n",
            "bedtools2/src/utils/htslib/hfile_libcurl.c\n",
            "bedtools2/src/utils/htslib/kstring.c\n",
            "bedtools2/src/utils/htslib/os/\n",
            "bedtools2/src/utils/htslib/md5.c\n",
            "bedtools2/src/utils/htslib/hfile_s3.c\n",
            "bedtools2/src/utils/htslib/htsfile.1\n",
            "bedtools2/src/utils/htslib/thread_pool.c\n",
            "bedtools2/src/utils/htslib/hfile.c\n",
            "bedtools2/src/utils/htslib/config.mk.in\n",
            "bedtools2/src/utils/htslib/tabix.c\n",
            "bedtools2/src/utils/htslib/textutils_internal.h\n",
            "bedtools2/src/utils/htslib/knetfile.c\n",
            "bedtools2/src/utils/htslib/bgzip.1\n",
            "bedtools2/src/utils/htslib/thread_pool_internal.h\n",
            "bedtools2/src/utils/htslib/os/rand.c\n",
            "bedtools2/src/utils/htslib/os/lzma_stub.h\n",
            "bedtools2/src/utils/htslib/m4/hts_prog_cc_warnings.m4\n",
            "bedtools2/src/utils/htslib/htslib/sam.h\n",
            "bedtools2/src/utils/htslib/htslib/synced_bcf_reader.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf_sweep.h\n",
            "bedtools2/src/utils/htslib/htslib/vcf.h\n",
            "bedtools2/src/utils/htslib/htslib/khash_str2int.h\n",
            "bedtools2/src/utils/htslib/htslib/bgzf.h\n",
            "bedtools2/src/utils/htslib/htslib/cram.h\n",
            "bedtools2/src/utils/htslib/htslib/kstring.h\n",
            "bedtools2/src/utils/htslib/htslib/knetfile.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_log.h\n",
            "bedtools2/src/utils/htslib/htslib/hfile.h\n",
            "bedtools2/src/utils/htslib/htslib/thread_pool.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_defs.h\n",
            "bedtools2/src/utils/htslib/htslib/khash.h\n",
            "bedtools2/src/utils/htslib/htslib/regidx.h\n",
            "bedtools2/src/utils/htslib/htslib/faidx.h\n",
            "bedtools2/src/utils/htslib/htslib/vcfutils.h\n",
            "bedtools2/src/utils/htslib/htslib/kseq.h\n",
            "bedtools2/src/utils/htslib/htslib/kfunc.h\n",
            "bedtools2/src/utils/htslib/htslib/tbx.h\n",
            "bedtools2/src/utils/htslib/htslib/ksort.h\n",
            "bedtools2/src/utils/htslib/htslib/kbitset.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_os.h\n",
            "bedtools2/src/utils/htslib/htslib/klist.h\n",
            "bedtools2/src/utils/htslib/htslib/hts.h\n",
            "bedtools2/src/utils/htslib/htslib/hts_endian.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.c\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.c\n",
            "bedtools2/src/utils/htslib/cram/mFILE.h\n",
            "bedtools2/src/utils/htslib/cram/cram_external.c\n",
            "bedtools2/src/utils/htslib/cram/cram.h\n",
            "bedtools2/src/utils/htslib/cram/misc.h\n",
            "bedtools2/src/utils/htslib/cram/files.c\n",
            "bedtools2/src/utils/htslib/cram/os.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.c\n",
            "bedtools2/src/utils/htslib/cram/cram_io.h\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.h\n",
            "bedtools2/src/utils/htslib/cram/mFILE.c\n",
            "bedtools2/src/utils/htslib/cram/cram_index.h\n",
            "bedtools2/src/utils/htslib/cram/sam_header.h\n",
            "bedtools2/src/utils/htslib/cram/open_trace_file.h\n",
            "bedtools2/src/utils/htslib/cram/cram_codecs.c\n",
            "bedtools2/src/utils/htslib/cram/rANS_byte.h\n",
            "bedtools2/src/utils/htslib/cram/string_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/rANS_static.c\n",
            "bedtools2/src/utils/htslib/cram/pooled_alloc.h\n",
            "bedtools2/src/utils/htslib/cram/cram_samtools.h\n",
            "bedtools2/src/utils/htslib/cram/cram_stats.c\n",
            "bedtools2/src/utils/htslib/cram/cram_encode.c\n",
            "bedtools2/src/utils/htslib/cram/cram_structs.h\n",
            "bedtools2/src/utils/htslib/cram/cram_io.c\n",
            "bedtools2/src/utils/htslib/cram/cram_decode.c\n",
            "bedtools2/src/utils/tabFile/tabFile.cpp\n",
            "bedtools2/src/utils/tabFile/tabFile.h\n",
            "bedtools2/src/utils/tabFile/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.h\n",
            "bedtools2/src/genomeCoverageBed/Makefile.frag\n",
            "bedtools2/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBed.h\n",
            "bedtools2/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "bedtools2/src/maskFastaFromBed/Makefile.frag\n",
            "bedtools2/src/regressTest/RegressTest.h\n",
            "bedtools2/src/regressTest/RegressTest.cpp\n",
            "bedtools2/src/regressTest/regressTestMain.cpp\n",
            "bedtools2/src/regressTest/Makefile.frag\n",
            "bedtools2/src/bedToIgv/bedToIgv.cpp\n",
            "bedtools2/src/bedToIgv/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.h\n",
            "bedtools2/src/nucBed/nucBedMain.cpp\n",
            "bedtools2/src/nucBed/LargeFileSupport.h\n",
            "bedtools2/src/nucBed/Makefile.frag\n",
            "bedtools2/src/nucBed/nucBed.cpp\n",
            "bedtools2/src/reldist/reldist.cpp\n",
            "bedtools2/src/reldist/Makefile.frag\n",
            "bedtools2/src/reldist/reldistMain.cpp\n",
            "bedtools2/src/reldist/reldist.h\n",
            "bedtools2/src/getOverlap/getOverlap.cpp\n",
            "bedtools2/src/getOverlap/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBedMain.cpp\n",
            "bedtools2/src/randomBed/randomBed.cpp\n",
            "bedtools2/src/randomBed/Makefile.frag\n",
            "bedtools2/src/randomBed/randomBed.h\n",
            "bedtools2/src/annotateBed/annotateBed.cpp\n",
            "bedtools2/src/annotateBed/annotateBed.h\n",
            "bedtools2/src/annotateBed/annotateMain.cpp\n",
            "bedtools2/src/annotateBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBed.h\n",
            "bedtools2/src/slopBed/slopBed.cpp\n",
            "bedtools2/src/slopBed/Makefile.frag\n",
            "bedtools2/src/slopBed/slopBedMain.cpp\n",
            "bedtools2/src/mapFile/mapHelp.cpp\n",
            "bedtools2/src/mapFile/mapFile.cpp\n",
            "bedtools2/src/mapFile/Makefile.frag\n",
            "bedtools2/src/mapFile/mapFile.h\n",
            "bedtools2/src/pairToPair/pairToPair.h\n",
            "bedtools2/src/pairToPair/pairToPairMain.cpp\n",
            "bedtools2/src/pairToPair/pairToPair.cpp\n",
            "bedtools2/src/pairToPair/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.cpp\n",
            "bedtools2/src/linksBed/Makefile.frag\n",
            "bedtools2/src/linksBed/linksBed.h\n",
            "bedtools2/src/linksBed/linksMain.cpp\n",
            "bedtools2/src/fisher/fisher.h\n",
            "bedtools2/src/fisher/kfunc.h\n",
            "bedtools2/src/fisher/kfunc.cpp\n",
            "bedtools2/src/fisher/Makefile.frag\n",
            "bedtools2/src/fisher/fisher.cpp\n",
            "bedtools2/src/fisher/fisherHelp.cpp\n",
            "bedtools2/src/expand/Makefile.frag\n",
            "bedtools2/src/expand/expand.cpp\n",
            "bedtools2/src/complementFile/complementHelp.cpp\n",
            "bedtools2/src/complementFile/complementFile.h\n",
            "bedtools2/src/complementFile/Makefile.frag\n",
            "bedtools2/src/complementFile/complementFile.cpp\n",
            "bedtools2/src/bedToBam/bedToBam.cpp\n",
            "bedtools2/src/bedToBam/Makefile.frag\n",
            "bedtools2/src/split/splitBed.h\n",
            "bedtools2/src/split/splitBed.cpp\n",
            "bedtools2/src/split/splitBedMain.cpp\n",
            "bedtools2/src/split/Makefile.frag\n",
            "bedtools2/src/sortBed/sortMain.cpp\n",
            "bedtools2/src/sortBed/sortBed.cpp\n",
            "bedtools2/src/sortBed/sortBed.h\n",
            "bedtools2/src/sortBed/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingHelp.cpp\n",
            "bedtools2/src/spacingFile/spacingFile.h\n",
            "bedtools2/src/spacingFile/Makefile.frag\n",
            "bedtools2/src/spacingFile/spacingFile.cpp\n",
            "bedtools2/data/gerp.chr1.bed.gz\n",
            "bedtools2/data/simpleRepeats.chr1.bed.gz\n",
            "bedtools2/data/aluY.chr1.bed.gz\n",
            "bedtools2/data/refseq.chr1.exons.bed.gz\n",
            "bedtools2/data/knownGene.hg18.chr21.short.bed\n",
            "bedtools2/data/knownGene.hg18.chr21.bed\n",
            "bedtools2/.github/workflows/\n",
            "bedtools2/.github/workflows/main.yml\n",
            "bedtools2/scripts/makeBashScripts.py\n",
            "bedtools2/scripts/vcfsort.sh\n",
            "bedtools2/genomes/human.hg38.genome\n",
            "bedtools2/genomes/mouse.mm10.genome\n",
            "bedtools2/genomes/mouse.mm8.genome\n",
            "bedtools2/genomes/human.hg18.genome\n",
            "bedtools2/genomes/human.hg19.genome\n",
            "bedtools2/genomes/mouse.mm9.genome\n",
            "bedtools2/docs/index.rst\n",
            "bedtools2/docs/Makefile\n",
            "bedtools2/docs/conf.py\n",
            "bedtools2/docs/_static/\n",
            "bedtools2/docs/content/\n",
            "bedtools2/docs/bedtools.swiss.png\n",
            "bedtools2/docs/build.sh\n",
            "bedtools2/docs/make.bat\n",
            "bedtools2/docs/bedtools.ai\n",
            "bedtools2/docs/templates/\n",
            "bedtools2/docs/themes/\n",
            "bedtools2/docs/bedtools.png\n",
            "bedtools2/docs/themes/rtd/\n",
            "bedtools2/docs/themes/rtd/theme.conf\n",
            "bedtools2/docs/themes/rtd/static/\n",
            "bedtools2/docs/themes/rtd/static/rtd.css_t\n",
            "bedtools2/docs/templates/page.html\n",
            "bedtools2/docs/templates/sidebar-intro.html\n",
            "bedtools2/docs/content/general-usage.rst\n",
            "bedtools2/docs/content/tools/\n",
            "bedtools2/docs/content/images/\n",
            "bedtools2/docs/content/quick-start.rst\n",
            "bedtools2/docs/content/example-usage.rst\n",
            "bedtools2/docs/content/overview.rst\n",
            "bedtools2/docs/content/advanced-usage.rst\n",
            "bedtools2/docs/content/history.rst\n",
            "bedtools2/docs/content/installation.rst\n",
            "bedtools2/docs/content/tips-and-tricks.rst\n",
            "bedtools2/docs/content/bedtools-suite.rst\n",
            "bedtools2/docs/content/faq.rst\n",
            "bedtools2/docs/content/related-tools.rst\n",
            "bedtools2/docs/content/images/memory-comparo.png\n",
            "bedtools2/docs/content/images/speed-comparo.png\n",
            "bedtools2/docs/content/images/fisher/\n",
            "bedtools2/docs/content/images/github-zip-button.png\n",
            "bedtools2/docs/content/images/galaxy-bedtools.png\n",
            "bedtools2/docs/content/images/bedtoolsMenu.png\n",
            "bedtools2/docs/content/images/tool-glyphs/\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/merge-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/getfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/map-speed-comparo.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-plot.png\n",
            "bedtools2/docs/content/images/tool-glyphs/subtract-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/flank-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/map-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/genomecov-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/cluster-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.svg\n",
            "bedtools2/docs/content/images/tool-glyphs/random-glyph.ai\n",
            "bedtools2/docs/content/images/tool-glyphs/complement-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/slop-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/window-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/shuffle-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/closest-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/intersect-glyph.png\n",
            "bedtools2/docs/content/images/tool-glyphs/barski_binding_site.png\n",
            "bedtools2/docs/content/images/fisher/res-genes.png\n",
            "bedtools2/docs/content/tools/expand.rst\n",
            "bedtools2/docs/content/tools/genomecov.rst\n",
            "bedtools2/docs/content/tools/makewindows.rst\n",
            "bedtools2/docs/content/tools/igv.rst\n",
            "bedtools2/docs/content/tools/overlap.rst\n",
            "bedtools2/docs/content/tools/nuc.rst\n",
            "bedtools2/docs/content/tools/fisher.rst\n",
            "bedtools2/docs/content/tools/cluster.rst\n",
            "bedtools2/docs/content/tools/pairtobed.rst\n",
            "bedtools2/docs/content/tools/shift.rst\n",
            "bedtools2/docs/content/tools/complement.rst\n",
            "bedtools2/docs/content/tools/bed12tobed6.rst\n",
            "bedtools2/docs/content/tools/map.rst\n",
            "bedtools2/docs/content/tools/tag.rst\n",
            "bedtools2/docs/content/tools/pairtopair.rst\n",
            "bedtools2/docs/content/tools/slop.rst\n",
            "bedtools2/docs/content/tools/sort.rst\n",
            "bedtools2/docs/content/tools/reldist.rst\n",
            "bedtools2/docs/content/tools/closest.rst\n",
            "bedtools2/docs/content/tools/bedtobam.rst\n",
            "bedtools2/docs/content/tools/maskfasta.rst\n",
            "bedtools2/docs/content/tools/bamtofastq.rst\n",
            "bedtools2/docs/content/tools/multicov.rst\n",
            "bedtools2/docs/content/tools/flank.rst\n",
            "bedtools2/docs/content/tools/random.rst\n",
            "bedtools2/docs/content/tools/merge.rst\n",
            "bedtools2/docs/content/tools/shuffle.rst\n",
            "bedtools2/docs/content/tools/window.rst\n",
            "bedtools2/docs/content/tools/bedpetobam.rst\n",
            "bedtools2/docs/content/tools/groupby.rst\n",
            "bedtools2/docs/content/tools/multiinter.rst\n",
            "bedtools2/docs/content/tools/bamtobed.rst\n",
            "bedtools2/docs/content/tools/getfasta.rst\n",
            "bedtools2/docs/content/tools/unionbedg.rst\n",
            "bedtools2/docs/content/tools/subtract.rst\n",
            "bedtools2/docs/content/tools/coverage.rst\n",
            "bedtools2/docs/content/tools/jaccard.rst\n",
            "bedtools2/docs/content/tools/links.rst\n",
            "bedtools2/docs/content/tools/annotate.rst\n",
            "bedtools2/docs/content/tools/intersect.rst\n",
            "bedtools2/docs/_static/labibi.css\n",
            "bedtools2/docs/_static/labibi.js\n",
            "bedtools2/test/split/\n",
            "bedtools2/test/bedtobam/\n",
            "bedtools2/test/cluster/\n",
            "bedtools2/test/expand/\n",
            "bedtools2/test/shuffle/\n",
            "bedtools2/test/fisher/\n",
            "bedtools2/test/complement/\n",
            "bedtools2/test/intersect/\n",
            "bedtools2/test/sort/\n",
            "bedtools2/test/subtract/\n",
            "bedtools2/test/merge/\n",
            "bedtools2/test/general/\n",
            "bedtools2/test/slop/\n",
            "bedtools2/test/reldist/\n",
            "bedtools2/test/shift/\n",
            "bedtools2/test/genomecov/\n",
            "bedtools2/test/jaccard/\n",
            "bedtools2/test/groupby/\n",
            "bedtools2/test/bamtofastq/\n",
            "bedtools2/test/bed12tobed6/\n",
            "bedtools2/test/getfasta/\n",
            "bedtools2/test/map/\n",
            "bedtools2/test/bigchroms/\n",
            "bedtools2/test/spacing/\n",
            "bedtools2/test/sample/\n",
            "bedtools2/test/flank/\n",
            "bedtools2/test/negativecontrol/\n",
            "bedtools2/test/multicov/\n",
            "bedtools2/test/closest/\n",
            "bedtools2/test/test.sh\n",
            "bedtools2/test/coverage/\n",
            "bedtools2/test/bamtobed/\n",
            "bedtools2/test/makewindows/\n",
            "bedtools2/test/makewindows/test-makewindows.sh\n",
            "bedtools2/test/makewindows/a.33bp.bed\n",
            "bedtools2/test/makewindows/a.20bp.bed\n",
            "bedtools2/test/makewindows/input.bed\n",
            "bedtools2/test/makewindows/a.19bp.bed\n",
            "bedtools2/test/bamtobed/two_blocks.sam\n",
            "bedtools2/test/bamtobed/three_blocks.sam\n",
            "bedtools2/test/bamtobed/test-bamtobed.sh\n",
            "bedtools2/test/bamtobed/numeric_tag.sam\n",
            "bedtools2/test/bamtobed/extra-long-header.sam.gz\n",
            "bedtools2/test/bamtobed/two_blocks_w_D.sam\n",
            "bedtools2/test/bamtobed/one_block.sam\n",
            "bedtools2/test/coverage/two_blocks.sam\n",
            "bedtools2/test/coverage/chr_0-100.bed\n",
            "bedtools2/test/coverage/three_blocks.bam.bai\n",
            "bedtools2/test/coverage/sam-w-del.sam\n",
            "bedtools2/test/coverage/three_blocks_nomatch.bed\n",
            "bedtools2/test/coverage/two_blocks.bam.bai\n",
            "bedtools2/test/coverage/three_blocks.sam\n",
            "bedtools2/test/coverage/one_block.bam.bai\n",
            "bedtools2/test/coverage/test-coverage.sh\n",
            "bedtools2/test/coverage/x.bed\n",
            "bedtools2/test/coverage/y.bed\n",
            "bedtools2/test/coverage/a.bed\n",
            "bedtools2/test/coverage/c.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bam\n",
            "bedtools2/test/coverage/sam-w-del.bam.bai\n",
            "bedtools2/test/coverage/b.bed\n",
            "bedtools2/test/coverage/three_blocks_match.bed\n",
            "bedtools2/test/coverage/one_block.sam\n",
            "bedtools2/test/coverage/three_blocks_match_1bp.bed\n",
            "bedtools2/test/closest/d_d1F.bed\n",
            "bedtools2/test/closest/d_q1.bed\n",
            "bedtools2/test/closest/d_iu.bed\n",
            "bedtools2/test/closest/d_d1R.bed\n",
            "bedtools2/test/closest/d_q2.bed\n",
            "bedtools2/test/closest/tfbs.bed\n",
            "bedtools2/test/closest/null_b.bed\n",
            "bedtools2/test/closest/bug281_a.medium.bed\n",
            "bedtools2/test/closest/null_c.bed\n",
            "bedtools2/test/closest/null_a.bed\n",
            "bedtools2/test/closest/d_id.bed\n",
            "bedtools2/test/closest/a2.bed\n",
            "bedtools2/test/closest/test-closest.sh\n",
            "bedtools2/test/closest/kclosest/\n",
            "bedtools2/test/closest/close-a.bed\n",
            "bedtools2/test/closest/b.names.bed\n",
            "bedtools2/test/closest/close-b.bed\n",
            "bedtools2/test/closest/strand-test-d.bed\n",
            "bedtools2/test/closest/b-one-bp-closer.bed\n",
            "bedtools2/test/closest/islands.2.bed\n",
            "bedtools2/test/closest/bug157_b.bed\n",
            "bedtools2/test/closest/shores.bed\n",
            "bedtools2/test/closest/strand-test-a.bed\n",
            "bedtools2/test/closest/strand-test-c.bed\n",
            "bedtools2/test/closest/bug157_a.bed\n",
            "bedtools2/test/closest/strand-test-b.bed\n",
            "bedtools2/test/closest/bug281_b.bed\n",
            "bedtools2/test/closest/dmr.bed\n",
            "bedtools2/test/closest/a.bed\n",
            "bedtools2/test/closest/b.bed\n",
            "bedtools2/test/closest/a.names.bed\n",
            "bedtools2/test/closest/mdb3.bed\n",
            "bedtools2/test/closest/mdb2.bed\n",
            "bedtools2/test/closest/b2.bed\n",
            "bedtools2/test/closest/mdb1.bed\n",
            "bedtools2/test/closest/mq1.bed\n",
            "bedtools2/test/closest/d.bed\n",
            "bedtools2/test/closest/d4.bed\n",
            "bedtools2/test/closest/sortAndNaming/\n",
            "bedtools2/test/closest/islands.bed\n",
            "bedtools2/test/closest/bug281_b.flip.bed\n",
            "bedtools2/test/closest/bug281_a.flip.medium.bed\n",
            "bedtools2/test/closest/d_d2R.bed\n",
            "bedtools2/test/closest/d_d2F.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/closest/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/closest/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/closest/sortAndNaming/sq1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/closest/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/closest/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/closest/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/closest/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/closest/kclosest/test-kclosest.sh\n",
            "bedtools2/test/closest/kclosest/bug471_b.bed\n",
            "bedtools2/test/closest/kclosest/bug471_a.bed\n",
            "bedtools2/test/closest/kclosest/d2.bed\n",
            "bedtools2/test/closest/kclosest/q2.bed\n",
            "bedtools2/test/closest/kclosest/d3.bed\n",
            "bedtools2/test/closest/kclosest/d1.bed\n",
            "bedtools2/test/closest/kclosest/q1.bed\n",
            "bedtools2/test/multicov/two_blocks.sam\n",
            "bedtools2/test/multicov/test-multicov.sh\n",
            "bedtools2/test/multicov/test-multi.bed\n",
            "bedtools2/test/multicov/test-multi.2.sam\n",
            "bedtools2/test/multicov/multicov.bed\n",
            "bedtools2/test/multicov/test-multi.sam\n",
            "bedtools2/test/multicov/one_block.sam\n",
            "bedtools2/test/negativecontrol/test-negativecontrol.sh\n",
            "bedtools2/test/flank/test-flank.sh\n",
            "bedtools2/test/flank/a.bed\n",
            "bedtools2/test/flank/tiny.genome\n",
            "bedtools2/test/sample/test-sample.sh\n",
            "bedtools2/test/sample/human.hg19.genome\n",
            "bedtools2/test/spacing/a.bed\n",
            "bedtools2/test/spacing/test-spacing.sh\n",
            "bedtools2/test/bigchroms/test-bigchroms.sh\n",
            "bedtools2/test/bigchroms/big4.bed\n",
            "bedtools2/test/bigchroms/make-big-chrom.py\n",
            "bedtools2/test/bigchroms/big4c.bed\n",
            "bedtools2/test/bigchroms/bigx.bed\n",
            "bedtools2/test/bigchroms/abig.bed\n",
            "bedtools2/test/map/values.bam\n",
            "bedtools2/test/map/.Rhistory\n",
            "bedtools2/test/map/a.vsorted.bed\n",
            "bedtools2/test/map/three_blocks_nomatch.bed\n",
            "bedtools2/test/map/b.vsorted.bed\n",
            "bedtools2/test/map/test.vcf\n",
            "bedtools2/test/map/values.bed\n",
            "bedtools2/test/map/test.gff\n",
            "bedtools2/test/map/values5.bed\n",
            "bedtools2/test/map/ivls.bed\n",
            "bedtools2/test/map/values4.bed\n",
            "bedtools2/test/map/t.bed\n",
            "bedtools2/test/map/ivls2.bed\n",
            "bedtools2/test/map/values3.bed\n",
            "bedtools2/test/map/.RData\n",
            "bedtools2/test/map/values2.bed\n",
            "bedtools2/test/map/d.bed\n",
            "bedtools2/test/map/genome\n",
            "bedtools2/test/map/three_blocks_match.bed\n",
            "bedtools2/test/map/test-map.sh\n",
            "bedtools2/test/map/bug262_a.bed\n",
            "bedtools2/test/map/bug262_b.bed\n",
            "bedtools2/test/map/fullFields.bam\n",
            "bedtools2/test/getfasta/test.iupac.bed\n",
            "bedtools2/test/getfasta/rna.fasta\n",
            "bedtools2/test/getfasta/t.fa\n",
            "bedtools2/test/getfasta/test.iupac.fa\n",
            "bedtools2/test/getfasta/blocks.bed\n",
            "bedtools2/test/getfasta/t_fH.fa.fai\n",
            "bedtools2/test/getfasta/t.bed\n",
            "bedtools2/test/getfasta/t.fa.fai\n",
            "bedtools2/test/getfasta/test-getfasta.sh\n",
            "bedtools2/test/getfasta/t_fH.fa\n",
            "bedtools2/test/getfasta/rna.bed\n",
            "bedtools2/test/bed12tobed6/two_blocks.bed\n",
            "bedtools2/test/bed12tobed6/bed6.bed\n",
            "bedtools2/test/bed12tobed6/three_blocks.bed\n",
            "bedtools2/test/bed12tobed6/test-bed12tobed6.sh\n",
            "bedtools2/test/bed12tobed6/one_blocks.bed\n",
            "bedtools2/test/bamtofastq/test-bamtofastq.sh\n",
            "bedtools2/test/bamtofastq/test2.sam\n",
            "bedtools2/test/bamtofastq/golden.fq\n",
            "bedtools2/test/bamtofastq/golden.fq2\n",
            "bedtools2/test/bamtofastq/test.sam\n",
            "bedtools2/test/groupby/gdc.bam\n",
            "bedtools2/test/groupby/values3_case.header.bed\n",
            "bedtools2/test/groupby/test-groupby.sh\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed\n",
            "bedtools2/test/groupby/a_vcfSVtest.vcf\n",
            "bedtools2/test/groupby/values3.7fields.header.bed\n",
            "bedtools2/test/groupby/values3.unmarked_header.bed.2\n",
            "bedtools2/test/groupby/values3.no_header.bed\n",
            "bedtools2/test/groupby/values3.header.bed\n",
            "bedtools2/test/groupby/noPosvalues.header.bed\n",
            "bedtools2/test/groupby/test.bed\n",
            "bedtools2/test/groupby/bug569_problem.txt\n",
            "bedtools2/test/jaccard/a645.bed\n",
            "bedtools2/test/jaccard/short.bed\n",
            "bedtools2/test/jaccard/bMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bed\n",
            "bedtools2/test/jaccard/c.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bam\n",
            "bedtools2/test/jaccard/b.bed\n",
            "bedtools2/test/jaccard/b645.bed\n",
            "bedtools2/test/jaccard/e.bed\n",
            "bedtools2/test/jaccard/test-jaccard.sh\n",
            "bedtools2/test/jaccard/long.bed\n",
            "bedtools2/test/jaccard/three_blocks_match.bed\n",
            "bedtools2/test/jaccard/aMixedStrands.bed\n",
            "bedtools2/test/jaccard/a.bam\n",
            "bedtools2/test/genomecov/two_blocks.sam\n",
            "bedtools2/test/genomecov/test_ref.fa.fai\n",
            "bedtools2/test/genomecov/test-genomecov.sh\n",
            "bedtools2/test/genomecov/chip.sam\n",
            "bedtools2/test/genomecov/y.bam\n",
            "bedtools2/test/genomecov/test_ref.fa\n",
            "bedtools2/test/genomecov/sam-w-del.sam\n",
            "bedtools2/test/genomecov/three_blocks_nomatch.bed\n",
            "bedtools2/test/genomecov/three_blocks.sam\n",
            "bedtools2/test/genomecov/empty.bam\n",
            "bedtools2/test/genomecov/y.bed\n",
            "bedtools2/test/genomecov/empty.cram\n",
            "bedtools2/test/genomecov/genome.txt\n",
            "bedtools2/test/genomecov/three_blocks_match.bed\n",
            "bedtools2/test/genomecov/pair-chip.sam\n",
            "bedtools2/test/genomecov/one_block.sam\n",
            "bedtools2/test/genomecov/three_blocks_match_1bp.bed\n",
            "bedtools2/test/shift/test-shift.sh\n",
            "bedtools2/test/shift/issue_807.bed\n",
            "bedtools2/test/shift/a.bed\n",
            "bedtools2/test/shift/b.bed\n",
            "bedtools2/test/shift/tiny.genome\n",
            "bedtools2/test/shift/huge.genome\n",
            "bedtools2/test/shift/issue_807.genomesize\n",
            "bedtools2/test/reldist/issue_711.a.bed\n",
            "bedtools2/test/reldist/issue_711.b.bed\n",
            "bedtools2/test/reldist/test-reldist.sh\n",
            "bedtools2/test/slop/a.bed\n",
            "bedtools2/test/slop/tiny.genome\n",
            "bedtools2/test/slop/test-slop.sh\n",
            "bedtools2/test/general/hg19.fa.fai\n",
            "bedtools2/test/general/empty.bed\n",
            "bedtools2/test/general/a.chromheader.bed\n",
            "bedtools2/test/general/test-general.sh\n",
            "bedtools2/test/general/non-empty.bed.gz\n",
            "bedtools2/test/general/t.bed\n",
            "bedtools2/test/general/a.trackheader.bed\n",
            "bedtools2/test/merge/bamCol8Mean.txt\n",
            "bedtools2/test/merge/mergeBugTest.bed\n",
            "bedtools2/test/merge/expFormat.bed\n",
            "bedtools2/test/merge/bamCol9Mean.txt\n",
            "bedtools2/test/merge/unsorted.bed\n",
            "bedtools2/test/merge/bug254_d.bed\n",
            "bedtools2/test/merge/bug254_e.bed\n",
            "bedtools2/test/merge/bamCol6Collapse.txt\n",
            "bedtools2/test/merge/bamCol3Collapse.txt\n",
            "bedtools2/test/merge/vcfSVtest.vcf\n",
            "bedtools2/test/merge/bug254_c.bed\n",
            "bedtools2/test/merge/bamCol11Collapse.txt\n",
            "bedtools2/test/merge/a.full.bed\n",
            "bedtools2/test/merge/a.bed\n",
            "bedtools2/test/merge/test-merge.sh\n",
            "bedtools2/test/merge/precisionTest.bed\n",
            "bedtools2/test/merge/b.bed\n",
            "bedtools2/test/merge/a.names.bed\n",
            "bedtools2/test/merge/testA.vcf\n",
            "bedtools2/test/merge/bamCol1Collapse.txt\n",
            "bedtools2/test/merge/precisionTest2.bed\n",
            "bedtools2/test/merge/bamCol10Collapse.txt\n",
            "bedtools2/test/merge/bamCol5Mean.txt\n",
            "bedtools2/test/merge/a.gff\n",
            "bedtools2/test/merge/bamCol7Collapse.txt\n",
            "bedtools2/test/merge/bamCol4Mean.txt\n",
            "bedtools2/test/merge/a.full.bam\n",
            "bedtools2/test/merge/mixedStrands.bed\n",
            "bedtools2/test/merge/fullFields.bam\n",
            "bedtools2/test/subtract/test-subtract.sh\n",
            "bedtools2/test/subtract/a.bed\n",
            "bedtools2/test/subtract/c.bed\n",
            "bedtools2/test/subtract/b.bed\n",
            "bedtools2/test/subtract/b2.bed\n",
            "bedtools2/test/subtract/d.bed\n",
            "bedtools2/test/sort/names.txt\n",
            "bedtools2/test/sort/a.bed\n",
            "bedtools2/test/sort/test-sort.sh\n",
            "bedtools2/test/sort/b.bed\n",
            "bedtools2/test/intersect/two_blocks.sam\n",
            "bedtools2/test/intersect/a_with_bothUnmapped.bam\n",
            "bedtools2/test/intersect/gdc.gff\n",
            "bedtools2/test/intersect/bug223_sv1_b.vcf\n",
            "bedtools2/test/intersect/bed3.bed\n",
            "bedtools2/test/intersect/bug150_a.bed\n",
            "bedtools2/test/intersect/test_ref.fa.fai\n",
            "bedtools2/test/intersect/x.bam\n",
            "bedtools2/test/intersect/exons.issue750.bed\n",
            "bedtools2/test/intersect/performanceTest.sh\n",
            "bedtools2/test/intersect/bed6.strand.bed\n",
            "bedtools2/test/intersect/queryUnmappedMateMappedCoordsInvalid.bam\n",
            "bedtools2/test/intersect/bug223_sv1_a.vcf\n",
            "bedtools2/test/intersect/bug150_b.bed\n",
            "bedtools2/test/intersect/headerOnly.vcf\n",
            "bedtools2/test/intersect/gdc_one.bed\n",
            "bedtools2/test/intersect/bed6.strand2.bed\n",
            "bedtools2/test/intersect/null_b.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader.bed\n",
            "bedtools2/test/intersect/human.hg19.vSort.genome\n",
            "bedtools2/test/intersect/gdc.bam\n",
            "bedtools2/test/intersect/bed4.bed\n",
            "bedtools2/test/intersect/chromOrderB.bed\n",
            "bedtools2/test/intersect/bedplus.bed\n",
            "bedtools2/test/intersect/test_ref.fa\n",
            "bedtools2/test/intersect/bed5.bed\n",
            "bedtools2/test/intersect/oneRecordNoNewline.bed\n",
            "bedtools2/test/intersect/aWithHeaderVsB.txt\n",
            "bedtools2/test/intersect/null_c.bed\n",
            "bedtools2/test/intersect/j1.bed\n",
            "bedtools2/test/intersect/b.issue311.vcf\n",
            "bedtools2/test/intersect/null_a.bed\n",
            "bedtools2/test/intersect/oneUnmapped.bam\n",
            "bedtools2/test/intersect/chromOrderA.bed\n",
            "bedtools2/test/intersect/bed6.bed\n",
            "bedtools2/test/intersect/issue_773_b.bed\n",
            "bedtools2/test/intersect/new_test-intersect.sh\n",
            "bedtools2/test/intersect/chromsOutOfOrder.bed\n",
            "bedtools2/test/intersect/jim.bed\n",
            "bedtools2/test/intersect/three_blocks_nomatch.bed\n",
            "bedtools2/test/intersect/b_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/bug223_f.vcf\n",
            "bedtools2/test/intersect/a.issue311.bed\n",
            "bedtools2/test/intersect/three_blocks.sam\n",
            "bedtools2/test/intersect/gdc_exp\n",
            "bedtools2/test/intersect/bug44_b.bed\n",
            "bedtools2/test/intersect/bug223_d.vcf\n",
            "bedtools2/test/intersect/bug223_e.vcf\n",
            "bedtools2/test/intersect/a_testZeroLen.bed\n",
            "bedtools2/test/intersect/a_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/x.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_bgzipped.bed.gz\n",
            "bedtools2/test/intersect/dosLineChar_a.bed\n",
            "bedtools2/test/intersect/bug44_a.vcf.gz\n",
            "bedtools2/test/intersect/b.cram\n",
            "bedtools2/test/intersect/y.bed\n",
            "bedtools2/test/intersect/test-intersect.sh\n",
            "bedtools2/test/intersect/a_vcfSVtest.vcf\n",
            "bedtools2/test/intersect/blocks.bed12\n",
            "bedtools2/test/intersect/recordsOutOfOrder.bed\n",
            "bedtools2/test/intersect/a.bed\n",
            "bedtools2/test/intersect/jim.vcf\n",
            "bedtools2/test/intersect/c.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bam\n",
            "bedtools2/test/intersect/dosLineCharWithExtraTab_b.bed\n",
            "bedtools2/test/intersect/a.cram\n",
            "bedtools2/test/intersect/b.bed\n",
            "bedtools2/test/intersect/two_blocks_partial.bam.bai\n",
            "bedtools2/test/intersect/foo\n",
            "bedtools2/test/intersect/split.issue750.bam\n",
            "bedtools2/test/intersect/mapped_and_unmapped.sam\n",
            "bedtools2/test/intersect/two_blocks_partial.bam\n",
            "bedtools2/test/intersect/e.bed\n",
            "bedtools2/test/intersect/a_withLargeHeader_gzipped.bed.gz\n",
            "bedtools2/test/intersect/svlen.vcf\n",
            "bedtools2/test/intersect/d.bed\n",
            "bedtools2/test/intersect/splitBug155_a.bed\n",
            "bedtools2/test/intersect/multi_intersect/\n",
            "bedtools2/test/intersect/notexthdr.bam\n",
            "bedtools2/test/intersect/large_b.bed\n",
            "bedtools2/test/intersect/nonamecheck_a.bed\n",
            "bedtools2/test/intersect/issue_773_y.bed\n",
            "bedtools2/test/intersect/sortAndNaming/\n",
            "bedtools2/test/intersect/split.issue750.sam\n",
            "bedtools2/test/intersect/a_gzipped.bed.gz\n",
            "bedtools2/test/intersect/splitBug155_b.bed\n",
            "bedtools2/test/intersect/bug167_strandSweep.bed\n",
            "bedtools2/test/intersect/large_a.bed\n",
            "bedtools2/test/intersect/nonamecheck_b.bed\n",
            "bedtools2/test/intersect/b.issue311.gff\n",
            "bedtools2/test/intersect/two_blocks_partial.bed\n",
            "bedtools2/test/intersect/three_blocks_match.bed\n",
            "bedtools2/test/intersect/human.hg19.genome\n",
            "bedtools2/test/intersect/bug187_a.bed\n",
            "bedtools2/test/intersect/strand_with_star.b.bed\n",
            "bedtools2/test/intersect/b_testZeroLen.bed\n",
            "bedtools2/test/intersect/one_block.sam\n",
            "bedtools2/test/intersect/bug187_b.bed\n",
            "bedtools2/test/intersect/three_blocks_match_1bp.bed\n",
            "bedtools2/test/intersect/strand_with_star.a.bed\n",
            "bedtools2/test/intersect/aVSb.bam\n",
            "bedtools2/test/intersect/a.bam\n",
            "bedtools2/test/intersect/sortAndNaming/num_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/db1_noChr.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_all.bed\n",
            "bedtools2/test/intersect/sortAndNaming/test-sort-and-naming.sh\n",
            "bedtools2/test/intersect/sortAndNaming/exp1\n",
            "bedtools2/test/intersect/sortAndNaming/q1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/num_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db2_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_leadingZero.txt\n",
            "bedtools2/test/intersect/sortAndNaming/sq1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db1_gls.bed\n",
            "bedtools2/test/intersect/sortAndNaming/alpha_missing.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/\n",
            "bedtools2/test/intersect/sortAndNaming/sdb1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/db3_numBackwards.bed\n",
            "bedtools2/test/intersect/sortAndNaming/q1a_num.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/q500K.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/dbchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_chr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/diffs.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/curr500Kvs500K.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/qchr1.bed\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_5v5.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/currChr1.txt\n",
            "bedtools2/test/intersect/sortAndNaming/bigTests/db500K.bed\n",
            "bedtools2/test/intersect/multi_intersect/query3.bed\n",
            "bedtools2/test/intersect/multi_intersect/query2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d8.bed\n",
            "bedtools2/test/intersect/multi_intersect/d9.bed\n",
            "bedtools2/test/intersect/multi_intersect/query.bed\n",
            "bedtools2/test/intersect/multi_intersect/g.bed\n",
            "bedtools2/test/intersect/multi_intersect/test-multi_intersect.sh\n",
            "bedtools2/test/intersect/multi_intersect/d4.bed\n",
            "bedtools2/test/intersect/multi_intersect/d5.bed\n",
            "bedtools2/test/intersect/multi_intersect/d7.bed\n",
            "bedtools2/test/intersect/multi_intersect/d6.bed\n",
            "bedtools2/test/intersect/multi_intersect/d2.bed\n",
            "bedtools2/test/intersect/multi_intersect/d3.bed\n",
            "bedtools2/test/intersect/multi_intersect/d1.bed\n",
            "bedtools2/test/complement/test-complement.sh\n",
            "bedtools2/test/complement/issue_503.genome\n",
            "bedtools2/test/complement/issue_503.bed\n",
            "bedtools2/test/fisher/test-fisher.sh\n",
            "bedtools2/test/fisher/sim.py\n",
            "bedtools2/test/fisher/plot.py\n",
            "bedtools2/test/fisher/cmp.sh\n",
            "bedtools2/test/fisher/dm6.fai\n",
            "bedtools2/test/fisher/t.60.genome\n",
            "bedtools2/test/fisher/get.sh\n",
            "bedtools2/test/fisher/README.md\n",
            "bedtools2/test/fisher/a.bed\n",
            "bedtools2/test/fisher/b.bed\n",
            "bedtools2/test/fisher/tumor.gff\n",
            "bedtools2/test/fisher/t.500.genome\n",
            "bedtools2/test/fisher/a_merge.bed\n",
            "bedtools2/test/fisher/shuf.sh\n",
            "bedtools2/test/fisher/test.bed\n",
            "bedtools2/test/shuffle/incl.sizedif.bed\n",
            "bedtools2/test/shuffle/test-shuffle.sh\n",
            "bedtools2/test/shuffle/excl.bed\n",
            "bedtools2/test/shuffle/simrep.bed\n",
            "bedtools2/test/shuffle/incl.bed\n",
            "bedtools2/test/expand/expand.txt\n",
            "bedtools2/test/expand/test-expand.sh\n",
            "bedtools2/test/cluster/in.bed\n",
            "bedtools2/test/cluster/test-cluster.sh\n",
            "bedtools2/test/bedtobam/test-bedtobam.sh\n",
            "bedtools2/test/bedtobam/chrsize.tmp\n",
            "bedtools2/test/split/test-split.sh\n",
            "bedtools2/test/split/randData.bed\n",
            "bedtools2/tutorial/answers.md\n",
            "bedtools2/tutorial/template.class.html\n",
            "bedtools2/tutorial/answers.html\n",
            "bedtools2/tutorial/bootstrap.css\n",
            "bedtools2/tutorial/Makefile\n",
            "bedtools2/tutorial/bedtools.html\n",
            "bedtools2/tutorial/bedtools.md\n",
            "bedtools2/tutorial/template.html\n",
            "Building BEDTools:\n",
            "=========================================================\n",
            "CXXFLAGS is [-g -Wall -O2 -std=c++11 ]\n",
            "DETECTED_VERSION = v2.30.0\n",
            "CURRENT_VERSION  = \n",
            "Updating version file.\n",
            "  * compiling src/bedtools.cpp\n",
            "  * compiling src/annotateBed/annotateMain.cpp\n",
            "  * compiling src/annotateBed/annotateBed.cpp\n",
            "  * compiling src/bamToBed/bamToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:88:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KuseAlignmentScore\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   88 |     bool \u001b[01;35m\u001b[KuseAlignmentScore\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToBed/bamToBed.cpp:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastqMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bamtofastq_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bamToFastq/bamToFastqMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFastq2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveFastq2\u001b[m\u001b[K    = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bamToFastq/bamToFastq.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/bamToFastq/bamToFastq.cpp:15\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToBam/bedToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedtobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedToBam/bedToBam.cpp:60:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   60 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedpeToBam/bedpeToBam.cpp\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint bedpetobam_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/bedpeToBam/bedpeToBam.cpp:61:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMapQual\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   61 |     bool \u001b[01;35m\u001b[KhaveMapQual\u001b[m\u001b[K     = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/bedToIgv/bedToIgv.cpp\n",
            "  * compiling src/bed12ToBed6/bed12ToBed6.cpp\n",
            "  * compiling src/closestFile/closestHelp.cpp\n",
            "  * compiling src/closestFile/closestFile.cpp\n",
            "  * compiling src/clusterBed/clusterMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint cluster_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/clusterBed/clusterMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveMaxDistance\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveMaxDistance\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/clusterBed/clusterBed.cpp\n",
            "  * compiling src/complementFile/complementHelp.cpp\n",
            "  * compiling src/complementFile/complementFile.cpp\n",
            "  * compiling src/coverageFile/coverageHelp.cpp\n",
            "  * compiling src/coverageFile/coverageFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvirtual void CoverageFile::giveFinalReport(RecordOutputMgr*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:100:11:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  100 |   \u001b[01;35m\u001b[Kasprintf(&depthPctString, \"%0.7f\", depthPct)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doMean(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:181:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  181 |  \u001b[01;35m\u001b[Kasprintf(&meanString, \"%0.7f\", mean)\u001b[m\u001b[K;\n",
            "      |  \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doHist(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:210:11:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  210 |   \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid CoverageFile::doDefault(RecordOutputMgr*, RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/coverageFile/coverageFile.cpp:230:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint asprintf(char**, const char*, ...)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  230 |  \u001b[01;35m\u001b[Kasprintf(&coveredFractionString, \"%0.7f\", coveredFraction)\u001b[m\u001b[K;\n",
            "      |  \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/expand/expand.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBedMain.cpp\n",
            "  * compiling src/fastaFromBed/fastaFromBed.cpp\n",
            "  * compiling src/flankBed/flankBedMain.cpp\n",
            "  * compiling src/flankBed/flankBed.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageMain.cpp\n",
            "  * compiling src/genomeCoverageBed/genomeCoverageBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/GenomeFile/NewGenomeFile.h:20\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.h:13\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/genomeCoverageBed/genomeCoverageBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/getOverlap/getOverlap.cpp\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint getoverlap_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/getOverlap/getOverlap.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveColumns\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveColumns\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/groupBy/groupByHelp.cpp\n",
            "  * compiling src/groupBy/groupBy.cpp\n",
            "  * compiling src/intersectFile/intersectHelp.cpp\n",
            "  * compiling src/intersectFile/intersectFile.cpp\n",
            "  * compiling src/fisher/fisherHelp.cpp\n",
            "  * compiling src/fisher/fisher.cpp\n",
            "  * compiling src/fisher/kfunc.cpp\n",
            "  * compiling src/jaccard/jaccardHelp.cpp\n",
            "  * compiling src/jaccard/jaccard.cpp\n",
            "  * compiling src/linksBed/linksMain.cpp\n",
            "  * compiling src/linksBed/linksBed.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
            "  * compiling src/maskFastaFromBed/maskFastaFromBed.cpp\n",
            "  * compiling src/mapFile/mapHelp.cpp\n",
            "  * compiling src/mapFile/mapFile.cpp\n",
            "  * compiling src/mergeFile/mergeHelp.cpp\n",
            "  * compiling src/mergeFile/mergeFile.cpp\n",
            "  * compiling src/multiBamCov/multiBamCovMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multibamcov_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:38:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBed\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   38 |     bool \u001b[01;35m\u001b[KhaveBed\u001b[m\u001b[K           = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:39:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveBams\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   39 |     bool \u001b[01;35m\u001b[KhaveBams\u001b[m\u001b[K          = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiBamCov/multiBamCovMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiBamCov/multiBamCov.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/BamReader.hpp:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamMultiReader.h:4\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.h:16\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/multiBamCov/multiBamCov.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint multiintersect_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:45:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   45 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:47:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   47 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/multiIntersectBed/multiIntersectBedMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/multiIntersectBed/multiIntersectBed.cpp\n",
            "  * compiling src/nucBed/nucBedMain.cpp\n",
            "  * compiling src/nucBed/nucBed.cpp\n",
            "  * compiling src/pairToBed/pairToBedMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtobed_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBedMain.cpp:43:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   43 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K       = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToBed/pairToBed.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid BedIntersectPE::FindSpanningOverlaps(const BEDPE&, std::vector<BED>&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:257:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "  257 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BedIntersectPE::FindOneOrMoreSpanningOverlaps(const BEDPE&, const string&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:310:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KspanLength\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "  310 |     CHRPOS \u001b[01;35m\u001b[KspanLength\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/pairToBed/pairToBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPairMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint pairtopair_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPairMain.cpp:44:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFraction\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   44 |     bool \u001b[01;35m\u001b[KhaveFraction\u001b[m\u001b[K = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/pairToPair/pairToPair.cpp\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid PairToPair::FindOverlaps(const BEDPE&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:112:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound1\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "  112 |         bool \u001b[01;35m\u001b[Kfound1\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/pairToPair/pairToPair.cpp:113:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kfound2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "  113 |         bool \u001b[01;35m\u001b[Kfound2\u001b[m\u001b[K = false;\n",
            "      |              \u001b[01;35m\u001b[K^~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/randomBed/randomBedMain.cpp\n",
            "  * compiling src/randomBed/randomBed.cpp\n",
            "  * compiling src/regressTest/regressTestMain.cpp\n",
            "  * compiling src/regressTest/RegressTest.cpp\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::executeAndCompareCorrectness(const fileListType&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:431:8:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  431 |  \u001b[01;35m\u001b[Ksystem(diffCmd.c_str())\u001b[m\u001b[K;\n",
            "      |  \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::startMemoryProfile(bool)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:573:7:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  573 |  \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |  \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool RegressTest::calcMemoryStats()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/regressTest/RegressTest.cpp:622:8:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kchar* fgets(char*, int, FILE*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  622 |   \u001b[01;35m\u001b[Kfgets(sLine, 4192, fp)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/reldist/reldistMain.cpp\n",
            "  * compiling src/reldist/reldist.cpp\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RelativeDistance::CalculateRelativeDistance()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/reldist/reldist.cpp:129:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[KCHRPOS\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  129 |             if (\u001b[01;35m\u001b[Klow_idx != chrom_mids->size() - 1\u001b[m\u001b[K)\n",
            "      |                 \u001b[01;35m\u001b[K~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/sampleFile/sampleHelp.cpp\n",
            "  * compiling src/sampleFile/sampleFile.cpp\n",
            "  * compiling src/shiftBed/shiftBedMain.cpp\n",
            "  * compiling src/shiftBed/shiftBed.cpp\n",
            "  * compiling src/shuffleBed/shuffleBedMain.cpp\n",
            "  * compiling src/shuffleBed/shuffleBed.cpp\n",
            "  * compiling src/slopBed/slopBedMain.cpp\n",
            "  * compiling src/slopBed/slopBed.cpp\n",
            "  * compiling src/sortBed/sortMain.cpp\n",
            "  * compiling src/sortBed/sortBed.cpp\n",
            "  * compiling src/spacingFile/spacingHelp.cpp\n",
            "  * compiling src/spacingFile/spacingFile.cpp\n",
            "  * compiling src/split/splitBedMain.cpp\n",
            "  * compiling src/split/splitBed.cpp\n",
            "  * compiling src/subtractFile/subtractHelp.cpp\n",
            "  * compiling src/subtractFile/subtractFile.cpp\n",
            "  * compiling src/summaryFile/summaryHelp.cpp\n",
            "  * compiling src/summaryFile/summaryFile.cpp\n",
            "  * compiling src/tagBam/tagBamMain.cpp\n",
            "  * compiling src/tagBam/tagBam.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.h:17\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/tagBam/tagBam.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphsMain.cpp\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint unionbedgraphs_main(int, char**)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:46:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiles\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   46 |     bool \u001b[01;35m\u001b[KhaveFiles\u001b[m\u001b[K         = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:48:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveGenome\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   48 |     bool \u001b[01;35m\u001b[KhaveGenome\u001b[m\u001b[K        = false;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/unionBedGraphs/unionBedGraphsMain.cpp:49:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[KhaveFiller\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   49 |     bool \u001b[01;35m\u001b[KhaveFiller\u001b[m\u001b[K        = true;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/unionBedGraphs/unionBedGraphs.cpp\n",
            "  * compiling src/windowBed/windowMain.cpp\n",
            "  * compiling src/windowBed/windowBed.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.h:15\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/windowBed/windowBed.cpp:13\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/windowMaker/windowMakerMain.cpp\n",
            "  * compiling src/windowMaker/windowMaker.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool BufferedStreamMgr::readFileChunk()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp:148:23:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  148 |   if (\u001b[01;35m\u001b[K_mainBufCurrLen < _useBufSize\u001b[m\u001b[K) {\n",
            "      |       \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~^~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
            "In file included from \u001b[01m\u001b[Ksrc/utils/BamTools/include/api/BamReader.h:1\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.h:14\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksrc/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp:8\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kint htslib_future::sam_hdr_rebuild(bam_hdr_t*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/BamTools/include/SamHeader.hpp:32:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint sam_hdr_write(samFile*, const bam_hdr_t*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "   32 |   \u001b[01;35m\u001b[Ksam_hdr_write(fp, hdr)\u001b[m\u001b[K;\n",
            "      |   \u001b[01;35m\u001b[K~~~~~~~~~~~~~^~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/Record.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/PlusFields.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BamRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint BlockMgr::getNonRedundantOverlap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/FileRecordTools/Records/BlockMgr.cpp:194:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Kstd::vector<long int>::size_type\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  194 |   for (int i = 1; \u001b[01;35m\u001b[Ki < _overlapStarts.size()\u001b[m\u001b[K; ++i)\n",
            "      |                   \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
            "  * compiling src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
            "  * compiling src/utils/bedFile/bedFile.cpp\n",
            "  * compiling src/utils/BinTree/BinTree.cpp\n",
            "  * compiling src/utils/version/version.cpp\n",
            "  * compiling src/utils/bedGraphFile/bedGraphFile.cpp\n",
            "  * compiling src/utils/chromsweep/chromsweep.cpp\n",
            "  * compiling src/utils/Contexts/ContextBase.cpp\n",
            "  * compiling src/utils/Contexts/ContextIntersect.cpp\n",
            "  * compiling src/utils/Contexts/ContextFisher.cpp\n",
            "  * compiling src/utils/Contexts/ContextMap.cpp\n",
            "  * compiling src/utils/Contexts/ContextSample.cpp\n",
            "  * compiling src/utils/Contexts/ContextSpacing.cpp\n",
            "  * compiling src/utils/Contexts/ContextMerge.cpp\n",
            "  * compiling src/utils/Contexts/ContextJaccard.cpp\n",
            "  * compiling src/utils/Contexts/ContextClosest.cpp\n",
            "  * compiling src/utils/Contexts/ContextSubtract.cpp\n",
            "  * compiling src/utils/Contexts/ContextCoverage.cpp\n",
            "  * compiling src/utils/Contexts/ContextComplement.cpp\n",
            "  * compiling src/utils/Contexts/ContextGroupBy.cpp\n",
            "  * compiling src/utils/Contexts/ContextSummary.cpp\n",
            "  * compiling src/utils/general/ParseTools.cpp\n",
            "  * compiling src/utils/general/PushBackStreamBuf.cpp\n",
            "  * compiling src/utils/general/CompressionTools.cpp\n",
            "  * compiling src/utils/general/Tokenizer.cpp\n",
            "  * compiling src/utils/general/CommonHelp.cpp\n",
            "  * compiling src/utils/general/ErrorMsg.cpp\n",
            "  * compiling src/utils/general/Random.cpp\n",
            "g++ -g -Wall -O2 -std=c++11  -Isrc/utils/gzstream  -c -o obj/gzstream.o src/utils/gzstream/gzstream.C\n",
            "  * compiling src/utils/fileType/fileType.cpp\n",
            "  * compiling src/utils/fileType/FileRecordTypeChecker.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kbool FileRecordTypeChecker::isTextDelimtedFormat(const char*, size_t)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/fileType/FileRecordTypeChecker.cpp:366:19:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  366 |   for (int j=0; \u001b[01;35m\u001b[Kj < len\u001b[m\u001b[K; j++) {\n",
            "      |                 \u001b[01;35m\u001b[K~~^~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/bedFilePE/bedFilePE.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOps.cpp\n",
            "  * compiling src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
            "  * compiling src/utils/NewChromsweep/NewChromsweep.cpp\n",
            "  * compiling src/utils/NewChromsweep/CloseSweep.cpp\n",
            "  * compiling src/utils/sequenceUtilities/sequenceUtils.cpp\n",
            "  * compiling src/utils/tabFile/tabFile.cpp\n",
            "  * compiling src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
            "  * compiling src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
            "  * compiling src/utils/Fasta/Fasta.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSequence(std::string)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:292:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  292 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kstd::string FastaReference::getSubSequence(std::string, CHRPOS, CHRPOS)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/Fasta/Fasta.cpp:334:14:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Ksize_t fread(void*, size_t, size_t, FILE*)\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
            "  334 |         \u001b[01;35m\u001b[Kfread(seq, sizeof(char), (off_t) seqlen, file)\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/Fasta/split.cpp\n",
            "  * compiling src/utils/VectorOps/VectorOps.cpp\n",
            "  * compiling src/utils/GenomeFile/NewGenomeFile.cpp\n",
            "  * compiling src/utils/GenomeFile/GenomeFile.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid GenomeFile::loadGenomeFileIntoMap()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/GenomeFile/GenomeFile.cpp:60:26:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kc2\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K-Wunused-but-set-variable\u001b[m\u001b[K]\n",
            "   60 |                     long \u001b[01;35m\u001b[Kc2\u001b[m\u001b[K;\n",
            "      |                          \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "  * compiling src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::printClosest(RecordKeyVector&, const std::vector<long int>*)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:170:42:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ and ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  170 |   if (\u001b[01;35m\u001b[Kcontext->getNumClosestHitsWanted() > keyList.size()\u001b[m\u001b[K)\n",
            "      |       \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid RecordOutputMgr::reportOverlapSummary(RecordKeyVector&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ksrc/utils/RecordOutputMgr/RecordOutputMgr.cpp:455:24:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcomparison of integer expressions of different signedness: ‘\u001b[01m\u001b[Ksize_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Klong unsigned int\u001b[m\u001b[K’} and ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K-Wsign-compare\u001b[m\u001b[K]\n",
            "  455 |   for (size_t i = 1; \u001b[01;35m\u001b[Ki < _context->getNumInputFiles()\u001b[m\u001b[K; i++)\n",
            "      |                      \u001b[01;35m\u001b[K~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "  * compiling src/utils/ToolBase/ToolBase.cpp\n",
            "  * compiling src/utils/driver/BedtoolsDriver.cpp\n",
            "cd src/utils/htslib && make config.h\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_FSEEKO 1' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "cd src/utils/htslib && make lib-static\n",
            "make[1]: Entering directory '/content/bedtools2/src/utils/htslib'\n",
            "gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"1.9\"' > version.h\n",
            "gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n",
            "gcc -g -Wall -O2 -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:495\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:53\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2095:9\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K-Wstringop-truncation\u001b[m\u001b[K]\n",
            "  106 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len, __bos (__dest))\u001b[m\u001b[K;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3748:13\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3823:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K-Wstringop-truncation\u001b[m\u001b[K]\n",
            "  106 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len, __bos (__dest))\u001b[m\u001b[K;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3740:9\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3823:17\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K-Wstringop-truncation\u001b[m\u001b[K]\n",
            "  106 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len, __bos (__dest))\u001b[m\u001b[K;\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n",
            "gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/bedtools2/src/utils/htslib'\n",
            "- Building main bedtools binary.\n",
            "done.\n",
            "- Creating executables for old CLI.\n",
            "done.\n",
            "  * compiling src/htsutil.cpp\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from collections import defaultdict\n",
        "from itertools import product # https://ru.wikipedia.org/wiki/%D0%9F%D1%80%D1%8F%D0%BC%D0%BE%D0%B5_%D0%BF%D1%80%D0%BE%D0%B8%D0%B7%D0%B2%D0%B5%D0%B4%D0%B5%D0%BD%D0%B8%D0%B5\n",
        "import matplotlib.pyplot as plt\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "import random\n",
        "import torch\n",
        "from torch import nn"
      ],
      "metadata": {
        "id": "f0kJbz7OJaav"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Set a random seed in a bunch of different places\n",
        "def set_seed(seed: int = 42) -> None:\n",
        "    np.random.seed(seed)\n",
        "    random.seed(seed)\n",
        "    torch.manual_seed(seed)\n",
        "    torch.cuda.manual_seed(seed)\n",
        "    # When running on the CuDNN backend, two further options must be set\n",
        "    torch.backends.cudnn.deterministic = True\n",
        "    torch.backends.cudnn.benchmark = False\n",
        "\n",
        "    print(f\"Random seed set as {seed}\")\n",
        "    \n",
        "set_seed(17)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3C1M0vp-JgzD",
        "outputId": "a4de3461-71c9-49f1-e64d-64f318d3f622"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Random seed set as 17\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "DEVICE = torch.device('cuda' if torch.cuda.is_available() else 'cpu')\n",
        "DEVICE"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZHKi5PtzJh7z",
        "outputId": "83d2b40b-b4f7-41a4-d784-50705223662d"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "device(type='cpu')"
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install gtfparse\n",
        "import gtfparse as gp"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "821UzA0y1-do",
        "outputId": "5cf02596-b2b1-41f6-e826-01e309db591b"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: gtfparse in /usr/local/lib/python3.10/dist-packages (2.0.1)\n",
            "Requirement already satisfied: polars in /usr/local/lib/python3.10/dist-packages (from gtfparse) (0.17.3)\n",
            "Requirement already satisfied: typing_extensions>=4.0.1 in /usr/local/lib/python3.10/dist-packages (from polars->gtfparse) (4.5.0)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Загрузка и подготовка данных"
      ],
      "metadata": {
        "id": "D3O6jGEaKEuF"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Загружаем геном и длины хромосом:"
      ],
      "metadata": {
        "id": "1uJpHn9fx9sN"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes\n",
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.knownGene.gtf.gz \n",
        "!gzip -d hg19.knownGene.gtf.gz"
      ],
      "metadata": {
        "id": "OWi1V-ixxl7p",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6461f8cf-44da-4d16-f7c2-f3cb0cd0526e"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-05-16 22:42:14--  https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1971 (1.9K)\n",
            "Saving to: ‘hg19.chrom.sizes’\n",
            "\n",
            "\rhg19.chrom.sizes      0%[                    ]       0  --.-KB/s               \rhg19.chrom.sizes    100%[===================>]   1.92K  --.-KB/s    in 0s      \n",
            "\n",
            "2023-05-16 22:42:14 (573 MB/s) - ‘hg19.chrom.sizes’ saved [1971/1971]\n",
            "\n",
            "--2023-05-16 22:42:14--  https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.knownGene.gtf.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 17960100 (17M) [application/x-gzip]\n",
            "Saving to: ‘hg19.knownGene.gtf.gz’\n",
            "\n",
            "hg19.knownGene.gtf. 100%[===================>]  17.13M  49.6MB/s    in 0.3s    \n",
            "\n",
            "2023-05-16 22:42:15 (49.6 MB/s) - ‘hg19.knownGene.gtf.gz’ saved [17960100/17960100]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Загружаем даннные разрывов. Используются данные ICGC 25-го релиза"
      ],
      "metadata": {
        "id": "UHYDZftR7FiA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/blood_all_data.csv\n",
        "!wget https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/brain_all_data.csv\n",
        "!wget https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/lungs_all_data_SBS.csv\n",
        "!wget https://github.com/xen234/cancer-genome-breakpoints-prediction/raw/main/data/raw/lungs_all_data_CNSM.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "r4qBa7WwnU68",
        "outputId": "f87ebb1a-f620-47c7-95bf-b87c9702749c"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-05-16 22:42:16--  https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/blood_all_data.csv\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 262410 (256K) [text/plain]\n",
            "Saving to: ‘blood_all_data.csv’\n",
            "\n",
            "blood_all_data.csv  100%[===================>] 256.26K  --.-KB/s    in 0.02s   \n",
            "\n",
            "2023-05-16 22:42:16 (12.1 MB/s) - ‘blood_all_data.csv’ saved [262410/262410]\n",
            "\n",
            "--2023-05-16 22:42:16--  https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/brain_all_data.csv\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 137950 (135K) [text/plain]\n",
            "Saving to: ‘brain_all_data.csv’\n",
            "\n",
            "brain_all_data.csv  100%[===================>] 134.72K  --.-KB/s    in 0.02s   \n",
            "\n",
            "2023-05-16 22:42:17 (7.97 MB/s) - ‘brain_all_data.csv’ saved [137950/137950]\n",
            "\n",
            "--2023-05-16 22:42:17--  https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/lungs_all_data_SBS.csv\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 245145 (239K) [text/plain]\n",
            "Saving to: ‘lungs_all_data_SBS.csv’\n",
            "\n",
            "lungs_all_data_SBS. 100%[===================>] 239.40K  --.-KB/s    in 0.02s   \n",
            "\n",
            "2023-05-16 22:42:17 (12.2 MB/s) - ‘lungs_all_data_SBS.csv’ saved [245145/245145]\n",
            "\n",
            "--2023-05-16 22:42:17--  https://github.com/xen234/cancer-genome-breakpoints-prediction/raw/main/data/raw/lungs_all_data_CNSM.zip\n",
            "Resolving github.com (github.com)... 192.30.255.112\n",
            "Connecting to github.com (github.com)|192.30.255.112|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/lungs_all_data_CNSM.zip [following]\n",
            "--2023-05-16 22:42:17--  https://raw.githubusercontent.com/xen234/cancer-genome-breakpoints-prediction/main/data/raw/lungs_all_data_CNSM.zip\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.111.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 8640858 (8.2M) [application/zip]\n",
            "Saving to: ‘lungs_all_data_CNSM.zip’\n",
            "\n",
            "lungs_all_data_CNSM 100%[===================>]   8.24M  --.-KB/s    in 0.06s   \n",
            "\n",
            "2023-05-16 22:42:18 (142 MB/s) - ‘lungs_all_data_CNSM.zip’ saved [8640858/8640858]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://raw.githubusercontent.com/KseniiaCheloshkina/cancer_breakpoints_hotspots_prediction/master/data/raw%20breakpoints/liver_all_data.csv"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0EnJ-s4XU7XN",
        "outputId": "9d1a7fee-0672-4b0e-9fe0-a911227034d4"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-05-16 22:42:18--  https://raw.githubusercontent.com/KseniiaCheloshkina/cancer_breakpoints_hotspots_prediction/master/data/raw%20breakpoints/liver_all_data.csv\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1665222 (1.6M) [text/plain]\n",
            "Saving to: ‘liver_all_data.csv’\n",
            "\n",
            "liver_all_data.csv  100%[===================>]   1.59M  --.-KB/s    in 0.03s   \n",
            "\n",
            "2023-05-16 22:42:18 (47.8 MB/s) - ‘liver_all_data.csv’ saved [1665222/1665222]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "lung_data = pd.read_csv('lungs_all_data_SBS.csv')\n",
        "blood_data = pd.read_csv('blood_all_data.csv')\n",
        "brain_data = pd.read_csv('brain_all_data.csv')"
      ],
      "metadata": {
        "id": "Spwq5DjE2NwT"
      },
      "execution_count": 26,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "liver_data = pd.read_csv('liver_all_data.csv')\n",
        "liver_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 485
        },
        "id": "uMEbRArsVB6w",
        "outputId": "607b7652-e800-4bfa-ef78-1929f0f925fa"
      },
      "execution_count": 61,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "      icgc_donor_id icgc_sample_id  \\\n",
              "0           DO50855       SA529897   \n",
              "1           DO50855       SA529897   \n",
              "2           DO50855       SA529897   \n",
              "3           DO50855       SA529897   \n",
              "4           DO50855       SA529897   \n",
              "...             ...            ...   \n",
              "22319       DO23508       SA270380   \n",
              "22320       DO23508       SA270380   \n",
              "22321       DO23508       SA270380   \n",
              "22322       DO23508       SA270380   \n",
              "22323       DO23508       SA270380   \n",
              "\n",
              "                                        variant_type chr   chr_bkpt  \\\n",
              "0                                           deletion   2   52981518   \n",
              "1                                 tandem duplication   4   35466414   \n",
              "2                                 tandem duplication   X   79729938   \n",
              "3                                           deletion  12   25209823   \n",
              "4      interchromosomal rearrangement - unknown type   7    7053667   \n",
              "...                                              ...  ..        ...   \n",
              "22319                                       deletion   1   52725289   \n",
              "22320                                       deletion   1  179907776   \n",
              "22321                                       deletion   9    6185918   \n",
              "22322                                      inversion   8  106335956   \n",
              "22323                                       deletion  10  114446497   \n",
              "\n",
              "       chr_range  chr_bkpt_beg  chr_bkpt_end  \n",
              "0              0      52981518      52981518  \n",
              "1              0      35466414      35466414  \n",
              "2              0      79729938      79729938  \n",
              "3              0      25209823      25209823  \n",
              "4              0       7053667       7053667  \n",
              "...          ...           ...           ...  \n",
              "22319          0      52725289      52725289  \n",
              "22320          0     179907776     179907776  \n",
              "22321          0       6185918       6185918  \n",
              "22322          0     106335956     106335956  \n",
              "22323          0     114446497     114446497  \n",
              "\n",
              "[22324 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-679d134f-7fe7-46aa-9496-6633f03aa822\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>icgc_donor_id</th>\n",
              "      <th>icgc_sample_id</th>\n",
              "      <th>variant_type</th>\n",
              "      <th>chr</th>\n",
              "      <th>chr_bkpt</th>\n",
              "      <th>chr_range</th>\n",
              "      <th>chr_bkpt_beg</th>\n",
              "      <th>chr_bkpt_end</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>2</td>\n",
              "      <td>52981518</td>\n",
              "      <td>0</td>\n",
              "      <td>52981518</td>\n",
              "      <td>52981518</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>4</td>\n",
              "      <td>35466414</td>\n",
              "      <td>0</td>\n",
              "      <td>35466414</td>\n",
              "      <td>35466414</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>X</td>\n",
              "      <td>79729938</td>\n",
              "      <td>0</td>\n",
              "      <td>79729938</td>\n",
              "      <td>79729938</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>12</td>\n",
              "      <td>25209823</td>\n",
              "      <td>0</td>\n",
              "      <td>25209823</td>\n",
              "      <td>25209823</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>7</td>\n",
              "      <td>7053667</td>\n",
              "      <td>0</td>\n",
              "      <td>7053667</td>\n",
              "      <td>7053667</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22319</th>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>1</td>\n",
              "      <td>52725289</td>\n",
              "      <td>0</td>\n",
              "      <td>52725289</td>\n",
              "      <td>52725289</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22320</th>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>1</td>\n",
              "      <td>179907776</td>\n",
              "      <td>0</td>\n",
              "      <td>179907776</td>\n",
              "      <td>179907776</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22321</th>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>9</td>\n",
              "      <td>6185918</td>\n",
              "      <td>0</td>\n",
              "      <td>6185918</td>\n",
              "      <td>6185918</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22322</th>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>inversion</td>\n",
              "      <td>8</td>\n",
              "      <td>106335956</td>\n",
              "      <td>0</td>\n",
              "      <td>106335956</td>\n",
              "      <td>106335956</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22323</th>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>10</td>\n",
              "      <td>114446497</td>\n",
              "      <td>0</td>\n",
              "      <td>114446497</td>\n",
              "      <td>114446497</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>22324 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-679d134f-7fe7-46aa-9496-6633f03aa822')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-679d134f-7fe7-46aa-9496-6633f03aa822 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-679d134f-7fe7-46aa-9496-6633f03aa822');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 61
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "lung_data.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 250
        },
        "id": "sJEhQki8qM2H",
        "outputId": "c7c47714-b4b7-4107-ae08-39aedcecb7f4"
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "  icgc_donor_id icgc_sample_id                                   variant_type  \\\n",
              "0       DO51591       SA534353  interchromosomal rearrangement - unknown type   \n",
              "1       DO51591       SA534353  interchromosomal rearrangement - unknown type   \n",
              "2       DO51591       SA534353  interchromosomal rearrangement - unknown type   \n",
              "3       DO51591       SA534353  interchromosomal rearrangement - unknown type   \n",
              "4       DO51591       SA534353  interchromosomal rearrangement - unknown type   \n",
              "\n",
              "  chr   chr_bkpt  chr_range  chr_bkpt_beg  chr_bkpt_end  \n",
              "0   1   84945093          0      84945093      84945093  \n",
              "1   1   84945093          0      84945093      84945093  \n",
              "2   3  179592240          0     179592240     179592240  \n",
              "3   3  179592240          0     179592240     179592240  \n",
              "4   3  179592240          0     179592240     179592240  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-93ba8406-ab35-4081-8770-932ae4b6b0fd\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>icgc_donor_id</th>\n",
              "      <th>icgc_sample_id</th>\n",
              "      <th>variant_type</th>\n",
              "      <th>chr</th>\n",
              "      <th>chr_bkpt</th>\n",
              "      <th>chr_range</th>\n",
              "      <th>chr_bkpt_beg</th>\n",
              "      <th>chr_bkpt_end</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>DO51591</td>\n",
              "      <td>SA534353</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>1</td>\n",
              "      <td>84945093</td>\n",
              "      <td>0</td>\n",
              "      <td>84945093</td>\n",
              "      <td>84945093</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>DO51591</td>\n",
              "      <td>SA534353</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>1</td>\n",
              "      <td>84945093</td>\n",
              "      <td>0</td>\n",
              "      <td>84945093</td>\n",
              "      <td>84945093</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>DO51591</td>\n",
              "      <td>SA534353</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>3</td>\n",
              "      <td>179592240</td>\n",
              "      <td>0</td>\n",
              "      <td>179592240</td>\n",
              "      <td>179592240</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>DO51591</td>\n",
              "      <td>SA534353</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>3</td>\n",
              "      <td>179592240</td>\n",
              "      <td>0</td>\n",
              "      <td>179592240</td>\n",
              "      <td>179592240</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>DO51591</td>\n",
              "      <td>SA534353</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>3</td>\n",
              "      <td>179592240</td>\n",
              "      <td>0</td>\n",
              "      <td>179592240</td>\n",
              "      <td>179592240</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-93ba8406-ab35-4081-8770-932ae4b6b0fd')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-93ba8406-ab35-4081-8770-932ae4b6b0fd button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-93ba8406-ab35-4081-8770-932ae4b6b0fd');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Скачали данные разрывов, данные легких также были обработаны, теперь нам нужны еще геномы, размеры хромосом и соответствующие омиксные данные"
      ],
      "metadata": {
        "id": "v5kWEBUcrTtB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!head hg19.chrom.sizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WaVNNUHamzfW",
        "outputId": "0ef585bc-2de0-43bb-bb11-8c460a7a8c94"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr1\t249250621\n",
            "chr2\t243199373\n",
            "chr3\t198022430\n",
            "chr4\t191154276\n",
            "chr5\t180915260\n",
            "chr6\t171115067\n",
            "chr7\t159138663\n",
            "chrX\t155270560\n",
            "chr8\t146364022\n",
            "chr9\t141213431\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "genome_df = pd.read_csv(\"hg19.knownGene.gtf\", sep='\\t', comment='#', header=None)"
      ],
      "metadata": {
        "id": "3SCquudVnFTL"
      },
      "execution_count": 45,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import gzip\n",
        "import shutil\n",
        "with gzip.open('copy_number_somatic_mutation.LUSC-US.tsv.gz', 'rb') as f_in:\n",
        "    with open('file.tsv', 'wb') as f_out:\n",
        "        shutil.copyfileobj(f_in, f_out)\n",
        "\n",
        "tsv_file='file.tsv'\n",
        "csv_table=pd.read_table(tsv_file,sep='\\t')\n",
        "csv_table.to_csv('lungs_second.csv',index=False)\n",
        "\n",
        "large_lung_data = pd.read_csv('lungs_second.csv')\n",
        "large_lung_data"
      ],
      "metadata": {
        "id": "FdRT8qnVbtlN"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Отсортируем геном"
      ],
      "metadata": {
        "id": "mxhQ7hnEDkIM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!grep -P \"^chr[0-9,X,Y,M][0-9]?\\t\" hg19.chrom.sizes > hg19.canon.genome\n",
        "!grep -P \"^chr[0-9,X,Y,M][0-9]?\\t\" hg19.knownGene.gtf > hg19.canon.gtf\n",
        "!head hg19.canon.gtf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uGCwdAsLRRq0",
        "outputId": "ab6a7a78-dc33-422b-d864-363cd7f6484f"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr1\tknownGene\ttranscript\t11874\t14409\t.\t+\t.\tgene_id \"uc001aaa.3\"; transcript_id \"uc001aaa.3\"; \n",
            "chr1\tknownGene\texon\t11874\t12227\t.\t+\t.\tgene_id \"uc001aaa.3\"; transcript_id \"uc001aaa.3\"; exon_number \"1\"; exon_id \"uc001aaa.3.1\";\n",
            "chr1\tknownGene\texon\t12613\t12721\t.\t+\t.\tgene_id \"uc001aaa.3\"; transcript_id \"uc001aaa.3\"; exon_number \"2\"; exon_id \"uc001aaa.3.2\";\n",
            "chr1\tknownGene\texon\t13221\t14409\t.\t+\t.\tgene_id \"uc001aaa.3\"; transcript_id \"uc001aaa.3\"; exon_number \"3\"; exon_id \"uc001aaa.3.3\";\n",
            "chr1\tknownGene\ttranscript\t11874\t14409\t.\t+\t.\tgene_id \"uc010nxr.1\"; transcript_id \"uc010nxr.1\"; \n",
            "chr1\tknownGene\texon\t11874\t12227\t.\t+\t.\tgene_id \"uc010nxr.1\"; transcript_id \"uc010nxr.1\"; exon_number \"1\"; exon_id \"uc010nxr.1.1\";\n",
            "chr1\tknownGene\texon\t12646\t12697\t.\t+\t.\tgene_id \"uc010nxr.1\"; transcript_id \"uc010nxr.1\"; exon_number \"2\"; exon_id \"uc010nxr.1.2\";\n",
            "chr1\tknownGene\texon\t13221\t14409\t.\t+\t.\tgene_id \"uc010nxr.1\"; transcript_id \"uc010nxr.1\"; exon_number \"3\"; exon_id \"uc010nxr.1.3\";\n",
            "chr1\tknownGene\ttranscript\t11874\t14409\t.\t+\t.\tgene_id \"B7ZGX9\"; transcript_id \"uc010nxq.1\";  gene_name \"B7ZGX9\";\n",
            "chr1\tknownGene\texon\t11874\t12227\t.\t+\t.\tgene_id \"B7ZGX9\"; transcript_id \"uc010nxq.1\"; exon_number \"1\"; exon_id \"uc010nxq.1.1\"; gene_name \"B7ZGX9\";\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools sort -i <(grep -P \"^chr[0-9,X,Y,M][0-9]?\\t\" hg19.knownGene.gtf) -g hg19.canon.genome > genome_sorted.gtf"
      ],
      "metadata": {
        "id": "iikG7loRRBJv"
      },
      "execution_count": 46,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "sorted_genome_df = pd.read_csv(\"genome_sorted.gtf\", sep='\\t', comment='#', header=None)\n",
        "sorted_genome_df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 380
        },
        "id": "4GBMETPoQdLS",
        "outputId": "0c568257-f986-4762-dccc-09aba7c8a1bb"
      },
      "execution_count": 50,
      "outputs": [
        {
          "output_type": "error",
          "ename": "ValueError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-50-93208d28d8d8>\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m sorted_genome_df = pd.read_csv(\"genome_sorted.gtf\", sep='\\t', comment='#', header=['Chromosome', 'Source', 'Feature', 'Start', 'End', 'Score', 'Strand', 'Frame', \n\u001b[0m\u001b[1;32m      2\u001b[0m                                                                                    'gene_id', 'transcript_id', 'exon_number', 'exon_id', 'gene_name'])\n\u001b[1;32m      3\u001b[0m \u001b[0msorted_genome_df\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    209\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    210\u001b[0m                     \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnew_arg_name\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_arg_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 211\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    212\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    213\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    329\u001b[0m                     \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfind_stack_level\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    330\u001b[0m                 )\n\u001b[0;32m--> 331\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    332\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    333\u001b[0m         \u001b[0;31m# error: \"Callable[[VarArg(Any), KwArg(Any)], Any]\" has no\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    948\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    949\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 950\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    951\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    952\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    603\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    604\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 605\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    606\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    607\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1432\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1433\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_file_or_buffer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1434\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_clean_options\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1435\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1436\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msqueeze\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"squeeze\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py\u001b[0m in \u001b[0;36m_clean_options\u001b[0;34m(self, options, engine)\u001b[0m\n\u001b[1;32m   1627\u001b[0m         \u001b[0mskiprows\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"skiprows\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1628\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1629\u001b[0;31m         \u001b[0mvalidate_header_arg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"header\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1630\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1631\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0marg\u001b[0m \u001b[0;32min\u001b[0m \u001b[0m_deprecated_defaults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/io/common.py\u001b[0m in \u001b[0;36mvalidate_header_arg\u001b[0;34m(header)\u001b[0m\n\u001b[1;32m    196\u001b[0m         \u001b[0mheader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mSequence\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    197\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mis_integer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 198\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"header must be integer or list of integers\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    199\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    200\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"cannot specify multi-index header with negative integers\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mValueError\u001b[0m: header must be integer or list of integers"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Фильтрация данных по геному"
      ],
      "metadata": {
        "id": "PY8P8qj6IMQB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install pybedtools\n",
        "from pybedtools import BedTool"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UTwABFH6OhCt",
        "outputId": "7ed0bbf3-2644-42f9-cec8-d3775258b476"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting pybedtools\n",
            "  Downloading pybedtools-0.9.0.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m101.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from pybedtools) (1.16.0)\n",
            "Collecting pysam (from pybedtools)\n",
            "  Downloading pysam-0.21.0-cp310-cp310-manylinux_2_24_x86_64.whl (20.0 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m20.0/20.0 MB\u001b[0m \u001b[31m39.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: cython in /usr/local/lib/python3.10/dist-packages (from pysam->pybedtools) (0.29.34)\n",
            "Building wheels for collected packages: pybedtools\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.9.0-cp310-cp310-linux_x86_64.whl size=14248391 sha256=2f1e9ef112720eba6f0ee72f81c18f7b8e6fcbf9bc385eb99ca5d5d2c900c4c7\n",
            "  Stored in directory: /root/.cache/pip/wheels/50/a7/83/8f6b841c60977cbd76f54f30f5b9e11159150d73a0a8fa42ff\n",
            "Successfully built pybedtools\n",
            "Installing collected packages: pysam, pybedtools\n",
            "Successfully installed pybedtools-0.9.0 pysam-0.21.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from natsort import natsorted \n",
        "distr=liver_data.groupby('chr')['icgc_sample_id'].nunique()\n",
        "\n",
        "distr.reindex(index = natsorted(distr.index)).plot(kind=\"bar\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 470
        },
        "id": "7qeitLIMVqRx",
        "outputId": "8a6bd412-ae08-4fa9-df35-e23b62cee430"
      },
      "execution_count": 20,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<Axes: xlabel='chr'>"
            ]
          },
          "metadata": {},
          "execution_count": 20
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 640x480 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAigAAAG0CAYAAAAYQdwgAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAy4ElEQVR4nO3de1hU9b7H8e9wG1ABBeW2RSDT1LyVKZEeQyUV3ZpptTVLTfMWasqplLbmpRLKTne37tpurXakdY5Z6ta9vaTUDi31kGVq3rUUdGsyXhJRvucPH+c0AcbCGfkNvl/Ps56HtX5rvvMdZWY+/NaaNTZVVQEAADCIT1U3AAAA8GsEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4/hVdQOVUVJSIocPH5bg4GCx2WxV3Q4AAKgAVZVTp05JTEyM+PhceY7EKwPK4cOHJTY2tqrbAAAAlXDo0CGpX7/+FffxyoASHBwsIpceYEhISBV3AwAAKsLhcEhsbKzzffxKvDKgXD6sExISQkABAMDLVOT0DE6SBQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABjHr6obcKf4ScsrvO/+rJ4e7AQAAFwNZlAAAIBxCCgAAMA4lgJKZmamtG3bVoKDgyUiIkL69OkjO3fudNnn3LlzkpaWJuHh4VKrVi3p16+fFBQUuOxz8OBB6dmzp9SoUUMiIiLkiSeekAsXLlz9owEAANWCpYCyfv16SUtLkw0bNsiqVaukuLhYunbtKmfOnHHuM2HCBFm6dKl8+OGHsn79ejl8+LD07dvXOX7x4kXp2bOnnD9/Xr744gt5++23ZcGCBfL000+771EBAACvZlNVreyNjx07JhEREbJ+/Xrp2LGjFBYWSr169SQ7O1vuvfdeERHZsWOHNG3aVHJzc+X222+XFStWyO9//3s5fPiwREZGiojI3LlzZeLEiXLs2DEJCAj4zft1OBwSGhoqhYWFEhIS4tzOSbIAAJirvPfvslzVOSiFhYUiIhIWFiYiIps3b5bi4mJJSUlx7tOkSRNp0KCB5ObmiohIbm6utGjRwhlORES6desmDodDtm3bVub9FBUVicPhcFkAAED1VemAUlJSIuPHj5f27dtL8+bNRUQkPz9fAgICpHbt2i77RkZGSn5+vnOfX4aTy+OXx8qSmZkpoaGhziU2NraybQMAAC9Q6YCSlpYm3377rSxcuNCd/ZQpIyNDCgsLncuhQ4c8fp8AAKDqVOpCbWPGjJFly5ZJTk6O1K9f37k9KipKzp8/LydPnnSZRSkoKJCoqCjnPl9++aVLvcuf8rm8z6/Z7Xax2+2VaRUAAHghSzMoqipjxoyRjz76SNauXSsJCQku423atBF/f39Zs2aNc9vOnTvl4MGDkpSUJCIiSUlJ8s0338jRo0ed+6xatUpCQkKkWbNmV/NYAABANWFpBiUtLU2ys7Pl448/luDgYOc5I6GhoRIUFCShoaEybNgwSU9Pl7CwMAkJCZGxY8dKUlKS3H777SIi0rVrV2nWrJk89NBD8sILL0h+fr5MnjxZ0tLSmCUBAAAiYjGgzJkzR0REkpOTXbbPnz9fhgwZIiIiL7/8svj4+Ei/fv2kqKhIunXrJn/605+c+/r6+sqyZctk9OjRkpSUJDVr1pTBgwfLjBkzru6RAACAauOqroNSVbgOCgAA3ueaXQcFAADAEwgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHL+qbsB08ZOWV2i//Vk9PdwJAADXD2ZQAACAcQgoAADAOAQUAABgHAIKAAAwjuWAkpOTI7169ZKYmBix2WyyZMkSl3GbzVbmMmvWLOc+8fHxpcazsrKu+sEAAIDqwfKneM6cOSOtWrWSoUOHSt++fUuNHzlyxGV9xYoVMmzYMOnXr5/L9hkzZsjw4cOd68HBwVZbAdyOT20BgBksB5TU1FRJTU0tdzwqKspl/eOPP5ZOnTrJDTfc4LI9ODi41L4AAAAiHj4HpaCgQJYvXy7Dhg0rNZaVlSXh4eFyyy23yKxZs+TChQvl1ikqKhKHw+GyAACA6sujF2p7++23JTg4uNShoHHjxsmtt94qYWFh8sUXX0hGRoYcOXJEXnrppTLrZGZmyvTp0z3ZKgAAMIhHA8pf//pXGThwoAQGBrpsT09Pd/7csmVLCQgIkJEjR0pmZqbY7fZSdTIyMlxu43A4JDY21nONAwCAKuWxgPLZZ5/Jzp07ZdGiRb+5b2Jioly4cEH2798vN910U6lxu91eZnABAADVk8cCyrx586RNmzbSqlWr39w3Ly9PfHx8JCIiwlPtVGsV/eSJCJ8+AQB4B8sB5fTp07J7927n+r59+yQvL0/CwsKkQYMGInLpEMyHH34o//Vf/1Xq9rm5ubJx40bp1KmTBAcHS25urkyYMEEefPBBqVOnzlU8FAAAUF1YDiibNm2STp06OdcvnxsyePBgWbBggYiILFy4UFRVBgwYUOr2drtdFi5cKNOmTZOioiJJSEiQCRMmuJxjUt1xrQ0AAK7MckBJTk4WVb3iPiNGjJARI0aUOXbrrbfKhg0brN4tAAC4jvBdPAAAwDgEFAAAYBwCCgAAMI5HL9QG78WJvACAqsQMCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4/ApHlwzfDIIAFBRzKAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYx3JAycnJkV69eklMTIzYbDZZsmSJy/iQIUPEZrO5LN27d3fZ58SJEzJw4EAJCQmR2rVry7Bhw+T06dNX9UAAAED1YTmgnDlzRlq1aiWzZ88ud5/u3bvLkSNHnMv777/vMj5w4EDZtm2brFq1SpYtWyY5OTkyYsQI690DAIBqyc/qDVJTUyU1NfWK+9jtdomKiipzbPv27bJy5Ur56quv5LbbbhMRkddff1169OghL774osTExFhtCQAAVDMeOQdl3bp1EhERITfddJOMHj1ajh8/7hzLzc2V2rVrO8OJiEhKSor4+PjIxo0by6xXVFQkDofDZQEAANWX2wNK9+7d5Z133pE1a9bI888/L+vXr5fU1FS5ePGiiIjk5+dLRESEy238/PwkLCxM8vPzy6yZmZkpoaGhziU2NtbdbQMAAINYPsTzW/r37+/8uUWLFtKyZUtp2LChrFu3Trp06VKpmhkZGZKenu5cdzgchBQAAKoxtweUX7vhhhukbt26snv3bunSpYtERUXJ0aNHXfa5cOGCnDhxotzzVux2u9jtdk+3Ci8TP2l5hffdn9XTg50AANzN49dB+eGHH+T48eMSHR0tIiJJSUly8uRJ2bx5s3OftWvXSklJiSQmJnq6HQAA4AUsz6CcPn1adu/e7Vzft2+f5OXlSVhYmISFhcn06dOlX79+EhUVJXv27JEnn3xSbrzxRunWrZuIiDRt2lS6d+8uw4cPl7lz50pxcbGMGTNG+vfvzyd4AACAiFQioGzatEk6derkXL98bsjgwYNlzpw5snXrVnn77bfl5MmTEhMTI127dpVnnnnG5RDNe++9J2PGjJEuXbqIj4+P9OvXT1577TU3PBwAlVHRw2UcKgNwrVgOKMnJyaKq5Y7/4x//+M0aYWFhkp2dbfWuAQDAdYLv4gEAAMYhoAAAAON4/GPGwPXO3ed38PFqANcDZlAAAIBxmEEB4BF8MgjA1WAGBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMbxq+oGAADXVvyk5RXab39WTw93ApSPGRQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEsB5ScnBzp1auXxMTEiM1mkyVLljjHiouLZeLEidKiRQupWbOmxMTEyKBBg+Tw4cMuNeLj48Vms7ksWVlZV/1gAABA9WD5Qm1nzpyRVq1aydChQ6Vv374uY2fPnpUtW7bIlClTpFWrVvLTTz/JY489Jr1795ZNmza57DtjxgwZPny4cz04OLiSDwHA9aCiFxcT4QJjQHVgOaCkpqZKampqmWOhoaGyatUql21vvPGGtGvXTg4ePCgNGjRwbg8ODpaoqCirdw8AAK4DHj8HpbCwUGw2m9SuXdtle1ZWloSHh8stt9wis2bNkgsXLpRbo6ioSBwOh8sCAACqL49+F8+5c+dk4sSJMmDAAAkJCXFuHzdunNx6660SFhYmX3zxhWRkZMiRI0fkpZdeKrNOZmamTJ8+3ZOtAgAAg3gsoBQXF8v9998vqipz5sxxGUtPT3f+3LJlSwkICJCRI0dKZmam2O32UrUyMjJcbuNwOCQ2NtZTrQMAgCrmkYByOZwcOHBA1q5d6zJ7UpbExES5cOGC7N+/X2666aZS43a7vczgAgAAqie3B5TL4WTXrl3y6aefSnh4+G/eJi8vT3x8fCQiIsLd7QAAAC9kOaCcPn1adu/e7Vzft2+f5OXlSVhYmERHR8u9994rW7ZskWXLlsnFixclPz9fRETCwsIkICBAcnNzZePGjdKpUycJDg6W3NxcmTBhgjz44INSp04d9z0yAADgtSwHlE2bNkmnTp2c65fPDRk8eLBMmzZNPvnkExERad26tcvtPv30U0lOTha73S4LFy6UadOmSVFRkSQkJMiECRNczjEBAADXN8sBJTk5WVS13PErjYmI3HrrrbJhwwardwsAAK4jfBcPAAAwDgEFAAAYh4ACAACMQ0ABAADG8eil7gEAQNWq6DeBm/Yt4MygAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4fBcPgOuWt35HCXA9YAYFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMI6f1Rvk5OTIrFmzZPPmzXLkyBH56KOPpE+fPs5xVZWpU6fKW2+9JSdPnpT27dvLnDlzpFGjRs59Tpw4IWPHjpWlS5eKj4+P9OvXT1599VWpVauWWx4UAMC7xU9aXuF992f19GAnqCqWZ1DOnDkjrVq1ktmzZ5c5/sILL8hrr70mc+fOlY0bN0rNmjWlW7ducu7cOec+AwcOlG3btsmqVatk2bJlkpOTIyNGjKj8owAAANWK5RmU1NRUSU1NLXNMVeWVV16RyZMny9133y0iIu+8845ERkbKkiVLpH///rJ9+3ZZuXKlfPXVV3LbbbeJiMjrr78uPXr0kBdffFFiYmJK1S0qKpKioiLnusPhsNo2AHglZhJwvbIcUK5k3759kp+fLykpKc5toaGhkpiYKLm5udK/f3/Jzc2V2rVrO8OJiEhKSor4+PjIxo0b5Z577ilVNzMzU6ZPn+7OVgEA15mKhj2CnhncepJsfn6+iIhERka6bI+MjHSO5efnS0REhMu4n5+fhIWFOff5tYyMDCksLHQuhw4dcmfbAADAMG6dQfEUu90udru9qtsAAADXiFtnUKKiokREpKCgwGV7QUGBcywqKkqOHj3qMn7hwgU5ceKEcx8AAHB9c2tASUhIkKioKFmzZo1zm8PhkI0bN0pSUpKIiCQlJcnJkydl8+bNzn3Wrl0rJSUlkpiY6M52AACAl7J8iOf06dOye/du5/q+ffskLy9PwsLCpEGDBjJ+/Hh59tlnpVGjRpKQkCBTpkyRmJgY57VSmjZtKt27d5fhw4fL3Llzpbi4WMaMGSP9+/cv8xM8AADg+mM5oGzatEk6derkXE9PTxcRkcGDB8uCBQvkySeflDNnzsiIESPk5MmT0qFDB1m5cqUEBgY6b/Pee+/JmDFjpEuXLs4Ltb322mtueDgAAKA6sBxQkpOTRVXLHbfZbDJjxgyZMWNGufuEhYVJdna21bsGAADXCb6LBwAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxLH+bMQAAvxY/aXmF9tuf1dPDnaC6YAYFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4fMwYANyIj9sC7sEMCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACM4/aAEh8fLzabrdSSlpYmIiLJycmlxkaNGuXuNgAAgBdz+5cFfvXVV3Lx4kXn+rfffit33XWX3Hfffc5tw4cPlxkzZjjXa9So4e42AACAF3N7QKlXr57LelZWljRs2FDuvPNO57YaNWpIVFSUu+8aAABUEx49B+X8+fPyt7/9TYYOHSo2m825/b333pO6detK8+bNJSMjQ86ePXvFOkVFReJwOFwWAABQfbl9BuWXlixZIidPnpQhQ4Y4tz3wwAMSFxcnMTExsnXrVpk4caLs3LlTFi9eXG6dzMxMmT59uidbBQAABvFoQJk3b56kpqZKTEyMc9uIESOcP7do0UKio6OlS5cusmfPHmnYsGGZdTIyMiQ9Pd257nA4JDY21nONAwCAKuWxgHLgwAFZvXr1FWdGREQSExNFRGT37t3lBhS73S52u93tPQIAADN5LKDMnz9fIiIipGfPnlfcLy8vT0REoqOjPdUKAABeIX7S8grvuz/ryu+v3s4jAaWkpETmz58vgwcPFj+//7+LPXv2SHZ2tvTo0UPCw8Nl69atMmHCBOnYsaO0bNnSE60AAAAv5JGAsnr1ajl48KAMHTrUZXtAQICsXr1aXnnlFTlz5ozExsZKv379ZPLkyZ5oAwAAeCmPBJSuXbuKqpbaHhsbK+vXr/fEXQIAgGqE7+IBAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBx/Kq6AQAAvFX8pOUV2m9/Vk8Pd1L9MIMCAACMQ0ABAADGIaAAAADjEFAAAIBx3B5Qpk2bJjabzWVp0qSJc/zcuXOSlpYm4eHhUqtWLenXr58UFBS4uw0AAODFPDKDcvPNN8uRI0ecy+eff+4cmzBhgixdulQ+/PBDWb9+vRw+fFj69u3riTYAAICX8sjHjP38/CQqKqrU9sLCQpk3b55kZ2dL586dRURk/vz50rRpU9mwYYPcfvvtnmgHAAB4GY/MoOzatUtiYmLkhhtukIEDB8rBgwdFRGTz5s1SXFwsKSkpzn2bNGkiDRo0kNzc3HLrFRUVicPhcFkAAED15faAkpiYKAsWLJCVK1fKnDlzZN++ffIf//EfcurUKcnPz5eAgACpXbu2y20iIyMlPz+/3JqZmZkSGhrqXGJjY93dNgAAMIjbD/GkpqY6f27ZsqUkJiZKXFycfPDBBxIUFFSpmhkZGZKenu5cdzgchBQAAKoxj3/MuHbt2tK4cWPZvXu3REVFyfnz5+XkyZMu+xQUFJR5zspldrtdQkJCXBYAAFB9eTygnD59Wvbs2SPR0dHSpk0b8ff3lzVr1jjHd+7cKQcPHpSkpCRPtwIAALyE2w/xPP7449KrVy+Ji4uTw4cPy9SpU8XX11cGDBggoaGhMmzYMElPT5ewsDAJCQmRsWPHSlJSEp/gAQAATm4PKD/88IMMGDBAjh8/LvXq1ZMOHTrIhg0bpF69eiIi8vLLL4uPj4/069dPioqKpFu3bvKnP/3J3W0AAAAv5vaAsnDhwiuOBwYGyuzZs2X27NnuvmsAAFBN8F08AADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBx3B5QMjMzpW3bthIcHCwRERHSp08f2blzp8s+ycnJYrPZXJZRo0a5uxUAAOCl3B5Q1q9fL2lpabJhwwZZtWqVFBcXS9euXeXMmTMu+w0fPlyOHDniXF544QV3twIAALyUn7sLrly50mV9wYIFEhERIZs3b5aOHTs6t9eoUUOioqLcffcAAKAa8Pg5KIWFhSIiEhYW5rL9vffek7p160rz5s0lIyNDzp49W26NoqIicTgcLgsAAKi+3D6D8kslJSUyfvx4ad++vTRv3ty5/YEHHpC4uDiJiYmRrVu3ysSJE2Xnzp2yePHiMutkZmbK9OnTPdkqAAAwiEcDSlpamnz77bfy+eefu2wfMWKE8+cWLVpIdHS0dOnSRfbs2SMNGzYsVScjI0PS09Od6w6HQ2JjYz3XOAAAqFIeCyhjxoyRZcuWSU5OjtSvX/+K+yYmJoqIyO7du8sMKHa7Xex2u0f6BAAA5nF7QFFVGTt2rHz00Ueybt06SUhI+M3b5OXliYhIdHS0u9sBAABeyO0BJS0tTbKzs+Xjjz+W4OBgyc/PFxGR0NBQCQoKkj179kh2drb06NFDwsPDZevWrTJhwgTp2LGjtGzZ0t3tAAAAL+T2gDJnzhwRuXQxtl+aP3++DBkyRAICAmT16tXyyiuvyJkzZyQ2Nlb69esnkydPdncrAADAS3nkEM+VxMbGyvr16919twAAoBrhu3gAAIBxCCgAAMA4BBQAAGAcAgoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAMA4BBQAAGIeAAgAAjENAAQAAxiGgAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACM41fVDQAAAO8SP2l5hfbbn9Wz0vfBDAoAADAOAQUAABiHgAIAAIxDQAEAAMYhoAAAAOMQUAAAgHEIKAAAwDgEFAAAYBwCCgAAME6VBpTZs2dLfHy8BAYGSmJionz55ZdV2Q4AADBElQWURYsWSXp6ukydOlW2bNkirVq1km7dusnRo0erqiUAAGCIKgsoL730kgwfPlwefvhhadasmcydO1dq1Kghf/3rX6uqJQAAYIgq+bLA8+fPy+bNmyUjI8O5zcfHR1JSUiQ3N7fU/kVFRVJUVORcLywsFBERh8Phsl9J0dkK9/Dr25anojUrWs8TNXnc174mj/va1+RxX/uaPO5rX7O6P+7L66r62zfWKvDjjz+qiOgXX3zhsv2JJ57Qdu3aldp/6tSpKiIsLCwsLCws1WA5dOjQb2aFKplBsSojI0PS09Od6yUlJXLixAkJDw8Xm81W7u0cDofExsbKoUOHJCQkxC29eENNb+jREzW9oUdP1PSGHj1R0xt69ERNb+jREzW9oUdP1PSGHq3UVFU5deqUxMTE/GbNKgkodevWFV9fXykoKHDZXlBQIFFRUaX2t9vtYrfbXbbVrl27wvcXEhLitv8Eb6rpDT16oqY39OiJmt7QoydqekOPnqjpDT16oqY39OiJmt7QY0VrhoaGVqhWlZwkGxAQIG3atJE1a9Y4t5WUlMiaNWskKSmpKloCAAAGqbJDPOnp6TJ48GC57bbbpF27dvLKK6/ImTNn5OGHH66qlgAAgCGqLKD84Q9/kGPHjsnTTz8t+fn50rp1a1m5cqVERka67T7sdrtMnTq11OGh6l7TG3r0RE1v6NETNb2hR0/U9IYePVHTG3r0RE1v6NETNb2hR0/VtKlW5LM+AAAA1w7fxQMAAIxDQAEAAMYhoAAAAOMQUAAAQIUcPnz4mt0XAaWa4txnAIC73XzzzZKdnX1N7ouAUk3Z7XbZvn17VbcBETly5Ig8/fTT0rlzZ2natKncfPPN0qtXL5k3b55cvHixqtsDUI1NmTJFLly4UO74wYMH5a677qpwveeee05Gjhwp9913n5w4ccIdLZbrugoohw4dkqFDh1q6zc8//yyff/65fPfdd6XGzp07J++8847lPrZv3y7z58+XHTt2iIjIjh07ZPTo0TJ06FBZu3atpVrp6ellLhcvXpSsrCzn+tU4c+aMzJ8/X/74xz/KG2+8IcePH7dcY8uWLbJv3z7n+rvvvivt27eX2NhY6dChgyxcuNByzbFjx8pnn31m+XZX8sYbb8igQYOc/bz77rvSrFkzadKkiTz11FNXfKKXZdOmTdK0aVP5+9//LsXFxbJr1y5p06aN1KxZUx5//HHp2LGjnDp1yq2PAUD5vvzyS3n11VclIyNDMjIy5NVXX5Uvv/zSI/f1008/Veo9oqSkpNztBw8etFTr7bfflrZt28q3335bauzPf/6zNG/eXPz8Kn5JtEcffVS2bt0qx48fl2bNmsnSpUst9WOJW76e2Evk5eWpj49PhfffuXOnxsXFqc1mUx8fH+3YsaMePnzYOZ6fn2+pnqrqihUrNCAgQMPCwjQwMFBXrFih9erV05SUFO3cubP6+vrqmjVrKlzPZrNp69atNTk52WWx2Wzatm1bTU5O1k6dOlnqsWnTpnr8+HFVVT148KDGx8draGiotm3bVsPCwjQiIkL37t1rqWbLli111apVqqr61ltvaVBQkI4bN07nzJmj48eP11q1aum8efMs1bz8/9KoUSPNysrSI0eOWLr9rz3zzDMaHBys/fr106ioKM3KytLw8HB99tlndebMmVqvXj19+umnLdVs3769Tps2zbn+7rvvamJioqqqnjhxQlu3bq3jxo2z3GtRUZEuWrRIx48fr/3799f+/fvr+PHj9YMPPtCioiLL9X5Lfn6+Tp8+3fLtDh06pKdOnSq1/fz587p+/XrL9f7973/r2rVrnb+fx44d06ysLJ0+fbp+9913luuVJyEhQb///vurrlNSUqJr167VN998U5cuXarnz5+3XOPQoUN67Ngx53pOTo4+8MAD2qFDBx04cGCpb4WviBdffFH3799v+XZXsnTpUp0yZYp+/vnnqqq6Zs0aTU1N1W7duumf//znStU8e/aszps3Tx9++GHt3r279ujRQ8eMGaOrV6+2XKugoEA7dOigNptN4+LitF27dtquXTvna3yHDh20oKCgUn2Wx+p7TmFhod53330aGBioEREROmXKFL1w4YJzvDLvOYWFhfrQQw+p3W7XmTNn6sWLF/XAgQPapUsXDQkJqfT/jarq66+/rn5+ftqiRQu95ZZbXBZ3qFYB5eOPP77i8vLLL1v6z+3Tp4/27NlTjx07prt27dKePXtqQkKCHjhwQFUr98uSlJSkf/zjH1VV9f3339c6deroU0895RyfNGmS3nXXXRWul5mZqQkJCaVCjZ+fn27bts1Sb5fZbDbnE3XgwIF6xx136MmTJ1VV9dSpU5qSkqIDBgywVDMoKMj5gnjLLbfom2++6TL+3nvvabNmzSz3uXr1an3ssce0bt266u/vr71799alS5fqxYsXLdVSVW3YsKH+z//8j6peemHx9fXVv/3tb87xxYsX64033mipZlBQkO7Zs8e5fvHiRfX399f8/HxVVf3nP/+pMTExlmru2rVLb7jhBg0MDNQ777xT77//fr3//vv1zjvv1MDAQL3xxht1165dlmr+FqsvtIcPH9a2bduqj4+P+vr66kMPPeQSVCrz3Nm4caOGhoaqzWbTOnXq6KZNmzQhIUEbNWqkDRs21KCgIN28ebOlmq+++mqZi6+vr2ZkZDjXKyo1NdX5XDl+/LgmJiaqzWbTevXqqY+PjzZp0kSPHj1qqcd27drp0qVLVVV1yZIl6uPjo71799aJEyfqPffco/7+/s7xirLZbOrr66spKSm6cOHCqw61c+fOVT8/P23Tpo2GhITou+++q8HBwfrII4/oyJEjNSgoSF955RVLNXft2qVxcXEaERGhsbGxarPZtGfPnpqYmKi+vr563333aXFxcYXr9evXT5OSknTHjh2lxnbs2KF33HGH3nvvvZZ6LCwsvOLy2WefWfo9HzdunDZu3Fg//PBDfeuttzQuLk579uzp/P/Jz89Xm81mqcfLlixZopGRkdqqVSsNCQnRlJSUqwqp+/fv106dOmm9evV08uTJOm3aNJfFHapVQLn8F7XNZit3sfLLEhERoVu3bnWul5SU6KhRo7RBgwa6Z8+eSr3IhoSEON88Ll68qH5+frplyxbn+DfffKORkZGWan755ZfauHFj/c///E/nX2juCig33HCD/vOf/3QZ/9e//qWxsbGWaoaHh+umTZtU9dK/a15ensv47t27NSgoqNJ9nj9/XhctWqTdunVTX19fjYmJ0aeeesrSG3VQUJAzfKqq+vv767fffutc379/v9aoUcNSj3Fxcc6/KFUvvXHbbDY9e/asqqru27dPAwMDLdVMSUnRu+++WwsLC0uNFRYW6t13361du3a1VPPrr7++4rJo0SJLv+uDBg3SxMRE/eqrr3TVqlXapk0bve222/TEiROqWrkX2pSUFH3kkUfU4XDorFmztH79+vrII484xx9++GHt06ePpZo2m03r16+v8fHxLovNZtPf/e53Gh8frwkJCZbqXf6dHD16tDZr1sw523jo0CFt06aNjho1ylKPNWvWdNZITEzUrKwsl/HXX3/d8l+sNptN58+fr3fffbf6+/treHi4PvbYY/rNN99YqnNZs2bNnH90rF27VgMDA3X27NnO8fnz52vTpk0t1UxNTdWRI0dqSUmJqqpmZWVpamqqqqp+//33Gh8fr1OnTq1wvVq1arm81v7apk2btFatWpZ6vPyeUt5i9T2nQYMG+umnnzrXjx07pu3atdOuXbvquXPnKvWec1l+fr6mpKSozWbTWrVq6bp16ypVR1X1zTff1ODgYL3nnnssB24rqlVAiYmJ0SVLlpQ7/r//+7+W/nODg4PLnDZOS0vT+vXra05OTqUCyu7du53rtWrVcvkLe//+/ZbfsFQvzWwMGjRIW7Zsqd988436+/tfVUC5/EsXExNT6kWrMj0++OCDOmzYMFVVve+++3Ty5Mku4zNnztQWLVpY7rOsKdkDBw7o1KlTNS4uztL/T0JCgq5YsUJVL70A+vj46AcffOAcX758ucbHx1vq8bHHHtPmzZvrihUrdO3atdqpUydNTk52jq9cuVIbNmxoqWZQUNAV30i2bt1aqbBXXrivzAttTEyMbty40bl+7tw57dWrl7Zu3VqPHz9eqRfaOnXqOJ+P58+fVx8fH5f72Lx5s/7ud7+zVHPkyJHaunXrUs/zygb8X/5O3nTTTfrxxx+7jK9evdpS4FFVDQ0N1a+//lpVL4X7yz9ftnv3bsvB+Zd9FhQU6PPPP69NmjRRHx8fbdu2rb755pvqcDgqXK+scP/L39F9+/ZZ7rFGjRouh9mKiorU399f//3vf6vqpRkBK8/H8PDwK74pf/rppxoeHm6px5CQEH3++ed13bp1ZS5vvfWWpd/zoKCgUofPHQ6HJiUlaefOnXXv3r2VCijZ2dkaFhamnTt31h07dugTTzyhAQEBOn78eP35558t1erWrZvWqVNH3377bct9WFWtAkqvXr10ypQp5Y7n5eVZ+qutbdu2+s4775Q5lpaWprVr17b8y9KyZUvnm6DqpRmTX05T5uTkWH4B+6X3339fIyMj1cfH56oCyuVjirVq1dL//u//dhlfv3695TeCH3/8UePj47Vjx46anp6uQUFB2qFDBx0+fLh27NhRAwICdPny5Zb7vNIx45KSklKzP1cyefJkrVevnj7yyCOakJCgkyZN0gYNGuicOXN07ty5GhsbqxMmTLDU46lTp/T+++9XPz8/tdlsescdd7i8AP3jH/9wCUEVER0dfcUp/U8++USjo6Mt1QwPD9d58+bp/v37y1yWL19u6Xe9Zs2apc7hKC4u1j59+mjLli1169atlp87NWvW1H379jnXfx3uDxw4UKlwv3jxYo2NjdXXX3/due1qAsrlcB8REeEyA6d6Kdzb7XZLNXv37q2TJk1S1UtvDr8+5PTWW29po0aNLPdZ1nMnJydHBw8erDVr1tSaNWtWuN7lP9hULz3XbTaby/N53bp1Wr9+fUs9xsTEuByy++mnn9RmszmD0969ey39Wz766KMaFxenixcvdpl9LCws1MWLF2t8fLyOGTPGUo/Jycn6/PPPlztu9T3npptuKvN18NSpU5qUlKStWrWy/Lzp27ev1qxZU1977TWX7f/617+0cePG2rhxY0vnMaWkpOihQ4cs9VBZ1Sqg5OTkuLz5/9rp06ctTWvNnDnTOaVYltGjR1uepp4zZ44uW7as3PGMjAznTENlHTp0SJcsWaKnT5+u1O1/fSxx5cqVLuOPP/649u/f33Ldn376SSdOnKjNmjXTwMBADQgI0Li4OH3ggQf0q6++slwvPj7e+deUO1y8eFGfe+45/f3vf68zZ87UkpISff/99zU2NlbDw8N1yJAhlf43/fnnn8s8WbQypkyZonXq1NGXXnpJv/76a83Pz9f8/Hz9+uuv9aWXXtKwsDBLU9+qql27dtVnnnmm3HGrL7QtWrQoFWxV/z+kNGjQwPILbZMmTVzOtVq2bJnzUJmq6oYNGyy/CV72ww8/aOfOnbV79+565MiRqwooPXr00HvuuUfr1KlTKkhu2LDB8iHc7777TsPDw3XQoEH6zDPPaK1atfTBBx/U5557TgcNGqR2u13nz59vqaaPj88Vw31hYWGp88SuJC0tTRs1aqTPPvustmvXTgcPHqxNmjTRFStW6MqVK7VFixY6dOhQSz0OHjxY77zzTt2+fbvu3btX//CHP7gcylq3bp2lQ83nzp3TUaNGaUBAgPr4+GhgYKAGBgaqj4+PBgQE6OjRo/XcuXOWenzzzTeveI5Sfn6+pfMxxo4dW+55MA6HQxMTEy0/b+64445yT/g+e/asjhs3Tv39/S3VvFaqVUABrhdZWVkaHR3tcgzcZrNpdHT0Ff+iK8/ixYv13XffLXf8xIkTumDBggrXe/LJJ8s9D6a4uFh79+5tOdxPmzZN33///XLHn3rqKe3bt6+lmr9UUlKiM2fO1KioKPX19a1UQBkyZIjLsmjRIpfxJ554Qrt162a57u7du7V///4aHBzsPPTm7++vd9xxh3700UeW6/3W7KNVp0+f1uHDh2vz5s11xIgRWlRUpLNmzdKAgAC12WyanJxs+f4KCgr09ttvd/6Ox8XFuZxD8uGHH5aaFaiIwsJCXbt2rWZnZ2t2drauXbu2zPO5qsKJEydKzbr9ksPhsHzuSEU+MFCZT9RdCzZVLjkKeKt9+/ZJfn6+iIhERUVJQkJCFXd0yYULF+Ts2bMSEhJS7viPP/4ocXFxbrvPs2fPiq+vr9jt9quqs3nzZvn8889l0KBBUqdOHTd1d8mZM2fE19dXAgMDK3V7VZWjR49KSUmJ1K1bV/z9/d3an7udO3dOiouLJTg4uNI1du3aJUVFRdKkSRNL1+uA97uuLtQGVDcJCQmSlJQkSUlJznBSmQsS/harNf38/MoNJyKXrq47ffp0d7TmdPz4cRk9evRV12nTpo089thjUqdOHbf/W544cUIeffTRSt/eZrNJZGSkREdHO8OJCf/f5QkMDJTg4OCrqteoUaMyLyZmyoU33V3TEz16rSqewQHgZlavWVIVNb2hR0/U9IYePVHThB7LuvDmjz/+6ByvzCfL3H0xT09cHNSbMV8GeJlPPvnkiuN79+6t8pre0KMnanpDj56o6Q09Tpw4UZo3by6bNm2SkydPyvjx46VDhw6ybt06adCggeX+yqvZvn37Std0dz1vxzkogJfx8fERm812xW+sttlslr6I0N01vaFHT9T0hh49UdMbeoyMjJTVq1dLixYtROTS+TyPPvqo/P3vf5dPP/1UatasKTExMZZ6dHdNT/TozTgHBfAy0dHRsnjxYikpKSlz2bJlS5XX9IYePVHTG3r0RE1v6PHnn392OY/FZrPJnDlzpFevXnLnnXfK999/b7lHd9f0RI/ejIACeJk2bdrI5s2byx3/rb86r0VNb+jREzW9oUdP1PSGHps0aSKbNm0qtf2NN96Qu+++W3r37m2pP0/U9ESPXu2ane0CwC3cfUFCT9T0hh49UdMbevRETW/o0RMX3nR3TU/06M04BwUAABiHQzwAAMA4BBQAAGAcAgoAADAOAQUAABiHgALACPv37xebzSZ5eXlV3QoAAxBQAACAcQgoALzWxYsXpaSkpKrbAOABBBQA11RJSYm88MILcuONN4rdbpcGDRrIc8895xzfu3evdOrUSWrUqCGtWrWS3Nxc59iCBQukdu3a8sknn0izZs3EbrfLwYMHq+JhAPAwAgqAayojI0OysrJkypQp8t1330l2drZERkY6x//4xz/K448/Lnl5edK4cWMZMGCAXLhwwTl+9uxZef755+Uvf/mLbNu2TSIiIqriYQDwMK4kC+CaOXXqlNSrV0/eeOMNeeSRR1zG9u/fLwkJCfKXv/xFhg0bJiIi3333ndx8882yfft2adKkiSxYsEAefvhhycvLk1atWlXFQwBwjTCDAuCa2b59uxQVFUmXLl3K3adly5bOn6Ojo0VE5OjRo85tAQEBLvsAqJ4IKACumaCgoN/cx9/f3/mzzWYTEXE5ETYoKMi5HUD1RUABcM00atRIgoKCZM2aNVXdCgDD+VV1AwCuH4GBgTJx4kR58sknJSAgQNq3by/Hjh2Tbdu2XfGwD4DrDwEFwDU1ZcoU8fPzk6effloOHz4s0dHRMmrUqKpuC4Bh+BQPAAAwDuegAAAA4xBQAACAcQgoAADAOAQUAABgHAIKAAAwDgEFAAAYh4ACAACMQ0ABAADGIaAAAADjEFAAAIBxCCgAAMA4/wf2deRkfJ5K2QAAAABJRU5ErkJggg==\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg19-blacklist.v2.bed.gz\n",
        "!gzip -d hg19-blacklist.v2.bed.gz\n",
        "!head hg19-blacklist.v2.bed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LpdfBAKGIpw1",
        "outputId": "b93fa9a5-fad6-4fe8-d6d7-b46c5386c21a"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-05-16 22:47:05--  https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg19-blacklist.v2.bed.gz\n",
            "Resolving github.com (github.com)... 192.30.255.113\n",
            "Connecting to github.com (github.com)|192.30.255.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://raw.githubusercontent.com/Boyle-Lab/Blacklist/master/lists/hg19-blacklist.v2.bed.gz [following]\n",
            "--2023-05-16 22:47:05--  https://raw.githubusercontent.com/Boyle-Lab/Blacklist/master/lists/hg19-blacklist.v2.bed.gz\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.109.133, 185.199.108.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 7731 (7.5K) [application/octet-stream]\n",
            "Saving to: ‘hg19-blacklist.v2.bed.gz’\n",
            "\n",
            "hg19-blacklist.v2.b 100%[===================>]   7.55K  --.-KB/s    in 0s      \n",
            "\n",
            "2023-05-16 22:47:06 (69.0 MB/s) - ‘hg19-blacklist.v2.bed.gz’ saved [7731/7731]\n",
            "\n",
            "chr10\t38726200\t42489100\tHigh Signal Region\n",
            "chr10\t42524900\t42819200\tHigh Signal Region\n",
            "chr10\t98560400\t98562500\tHigh Signal Region\n",
            "chr10\t135437600\t135534700\tHigh Signal Region\n",
            "chr11\t0\t196300\tHigh Signal Region\n",
            "chr11\t584400\t586500\tHigh Signal Region\n",
            "chr11\t964000\t966100\tLow Mappability\n",
            "chr11\t1015700\t1019100\tHigh Signal Region\n",
            "chr11\t1088800\t1094300\tHigh Signal Region\n",
            "chr11\t1141100\t1214300\tHigh Signal Region\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "liver_data.columns"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "gv8ncoI2v0rW",
        "outputId": "e1a7087f-44f3-4e15-836b-2cc819d21263"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Index(['icgc_donor_id', 'icgc_sample_id', 'variant_type', 'chr', 'chr_bkpt',\n",
              "       'chr_range', 'chr_bkpt_beg', 'chr_bkpt_end'],\n",
              "      dtype='object')"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "new_order = ['chr', 'chr_bkpt_beg', 'chr_bkpt_end', 'icgc_donor_id', 'icgc_sample_id', 'variant_type', 'chr_bkpt',\n",
        "       'chr_range']\n",
        "liver_data = liver_data[new_order]\n",
        "liver_data['chr'] = liver_data['chr'].apply(lambda x: 'chr' + x)\n",
        "liver_data.head()"
      ],
      "metadata": {
        "id": "nQulG-BxcQcE",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 250
        },
        "outputId": "1fddd1cc-d889-477c-b3ba-af7817943cd4"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "     chr  chr_bkpt_beg  chr_bkpt_end icgc_donor_id icgc_sample_id  \\\n",
              "0   chr2      52981518      52981518       DO50855       SA529897   \n",
              "1   chr4      35466414      35466414       DO50855       SA529897   \n",
              "2   chrX      79729938      79729938       DO50855       SA529897   \n",
              "3  chr12      25209823      25209823       DO50855       SA529897   \n",
              "4   chr7       7053667       7053667       DO50855       SA529897   \n",
              "\n",
              "                                    variant_type  chr_bkpt  chr_range  \n",
              "0                                       deletion  52981518          0  \n",
              "1                             tandem duplication  35466414          0  \n",
              "2                             tandem duplication  79729938          0  \n",
              "3                                       deletion  25209823          0  \n",
              "4  interchromosomal rearrangement - unknown type   7053667          0  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-516aacf7-9b31-4bf8-a5ae-10d3cbef6764\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>chr_bkpt_beg</th>\n",
              "      <th>chr_bkpt_end</th>\n",
              "      <th>icgc_donor_id</th>\n",
              "      <th>icgc_sample_id</th>\n",
              "      <th>variant_type</th>\n",
              "      <th>chr_bkpt</th>\n",
              "      <th>chr_range</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr2</td>\n",
              "      <td>52981518</td>\n",
              "      <td>52981518</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>52981518</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr4</td>\n",
              "      <td>35466414</td>\n",
              "      <td>35466414</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>35466414</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chrX</td>\n",
              "      <td>79729938</td>\n",
              "      <td>79729938</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>79729938</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr12</td>\n",
              "      <td>25209823</td>\n",
              "      <td>25209823</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>25209823</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr7</td>\n",
              "      <td>7053667</td>\n",
              "      <td>7053667</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>interchromosomal rearrangement - unknown type</td>\n",
              "      <td>7053667</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-516aacf7-9b31-4bf8-a5ae-10d3cbef6764')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-516aacf7-9b31-4bf8-a5ae-10d3cbef6764 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-516aacf7-9b31-4bf8-a5ae-10d3cbef6764');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "breakpoints = BedTool.from_dataframe(liver_data)\n",
        "filtered_breakpoints = breakpoints.intersect(\"hg19-blacklist.v2.bed\", v=True)"
      ],
      "metadata": {
        "id": "4d2oWTbrbYML"
      },
      "execution_count": 31,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "sorted_genome_df = pd.DataFrame(sorted_genome_df)\n",
        "sorted_genome_df.drop([1, 2], inplace=True, axis=1)\n",
        "sorted_genome = BedTool.from_dataframe(sorted_genome_df)\n",
        "filtered_breakpoints = filtered_breakpoints.intersect(sorted_genome, u=True)"
      ],
      "metadata": {
        "id": "u0kLHNKCcAXn"
      },
      "execution_count": 51,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "filtered_breakpoints.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "E_nyp6ll4DgA",
        "outputId": "a3e216df-2348-4bb3-90ea-30415c8f6355"
      },
      "execution_count": 53,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "chr12\t25209823\t25209823\tDO50855\tSA529897\tdeletion\t25209823\t0\n",
            " chr11\t89082611\t89082611\tDO50855\tSA529897\ttandem duplication\t89082611\t0\n",
            " chr20\t15231294\t15231294\tDO50855\tSA529897\ttandem duplication\t15231294\t0\n",
            " chr2\t2174329\t2174329\tDO50855\tSA529897\ttandem duplication\t2174329\t0\n",
            " chr2\t182905785\t182905785\tDO50855\tSA529897\tdeletion\t182905785\t0\n",
            " chr11\t2652987\t2652987\tDO50855\tSA529897\tdeletion\t2652987\t0\n",
            " chr6\t75852481\t75852481\tDO50855\tSA529897\tinterchromosomal rearrangement - unknown type\t75852481\t0\n",
            " chr11\t1745192\t1745192\tDO50855\tSA529897\tinversion\t1745192\t0\n",
            " chr12\t23254322\t23254322\tDO50855\tSA529897\ttandem duplication\t23254322\t0\n",
            " chr6\t69906443\t69906443\tDO50855\tSA529897\tdeletion\t69906443\t0\n",
            " "
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filtered_liver_df = filtered_breakpoints.to_dataframe()\n",
        "filtered_liver_df.columns = new_order\n",
        "filtered_liver_df"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 423
        },
        "id": "4Y8_VBf84O2C",
        "outputId": "e1d010a5-02b9-46d6-b4c2-5013ed6c3a72"
      },
      "execution_count": 60,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "         chr  chr_bkpt_beg  chr_bkpt_end icgc_donor_id icgc_sample_id  \\\n",
              "0      chr12      25209823      25209823       DO50855       SA529897   \n",
              "1      chr11      89082611      89082611       DO50855       SA529897   \n",
              "2      chr20      15231294      15231294       DO50855       SA529897   \n",
              "3       chr2       2174329       2174329       DO50855       SA529897   \n",
              "4       chr2     182905785     182905785       DO50855       SA529897   \n",
              "...      ...           ...           ...           ...            ...   \n",
              "11347   chr2     142085048     142085048       DO23508       SA270380   \n",
              "11348   chr2     211465699     211465699       DO23508       SA270380   \n",
              "11349   chr1      52725289      52725289       DO23508       SA270380   \n",
              "11350   chr8     106335956     106335956       DO23508       SA270380   \n",
              "11351  chr10     114446497     114446497       DO23508       SA270380   \n",
              "\n",
              "             variant_type   chr_bkpt  chr_range  \n",
              "0                deletion   25209823          0  \n",
              "1      tandem duplication   89082611          0  \n",
              "2      tandem duplication   15231294          0  \n",
              "3      tandem duplication    2174329          0  \n",
              "4                deletion  182905785          0  \n",
              "...                   ...        ...        ...  \n",
              "11347           inversion  142085048          0  \n",
              "11348  tandem duplication  211465699          0  \n",
              "11349            deletion   52725289          0  \n",
              "11350           inversion  106335956          0  \n",
              "11351            deletion  114446497          0  \n",
              "\n",
              "[11352 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-26428d49-5e24-4d39-ae3e-f1e9c0d4ee1d\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>chr</th>\n",
              "      <th>chr_bkpt_beg</th>\n",
              "      <th>chr_bkpt_end</th>\n",
              "      <th>icgc_donor_id</th>\n",
              "      <th>icgc_sample_id</th>\n",
              "      <th>variant_type</th>\n",
              "      <th>chr_bkpt</th>\n",
              "      <th>chr_range</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr12</td>\n",
              "      <td>25209823</td>\n",
              "      <td>25209823</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>25209823</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr11</td>\n",
              "      <td>89082611</td>\n",
              "      <td>89082611</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>89082611</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr20</td>\n",
              "      <td>15231294</td>\n",
              "      <td>15231294</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>15231294</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr2</td>\n",
              "      <td>2174329</td>\n",
              "      <td>2174329</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>2174329</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr2</td>\n",
              "      <td>182905785</td>\n",
              "      <td>182905785</td>\n",
              "      <td>DO50855</td>\n",
              "      <td>SA529897</td>\n",
              "      <td>deletion</td>\n",
              "      <td>182905785</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11347</th>\n",
              "      <td>chr2</td>\n",
              "      <td>142085048</td>\n",
              "      <td>142085048</td>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>inversion</td>\n",
              "      <td>142085048</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11348</th>\n",
              "      <td>chr2</td>\n",
              "      <td>211465699</td>\n",
              "      <td>211465699</td>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>tandem duplication</td>\n",
              "      <td>211465699</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11349</th>\n",
              "      <td>chr1</td>\n",
              "      <td>52725289</td>\n",
              "      <td>52725289</td>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>52725289</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11350</th>\n",
              "      <td>chr8</td>\n",
              "      <td>106335956</td>\n",
              "      <td>106335956</td>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>inversion</td>\n",
              "      <td>106335956</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11351</th>\n",
              "      <td>chr10</td>\n",
              "      <td>114446497</td>\n",
              "      <td>114446497</td>\n",
              "      <td>DO23508</td>\n",
              "      <td>SA270380</td>\n",
              "      <td>deletion</td>\n",
              "      <td>114446497</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11352 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-26428d49-5e24-4d39-ae3e-f1e9c0d4ee1d')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-26428d49-5e24-4d39-ae3e-f1e9c0d4ee1d button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-26428d49-5e24-4d39-ae3e-f1e9c0d4ee1d');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 60
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "filtered_liver_df.to_csv('filtered_liver_data.csv', index=False)"
      ],
      "metadata": {
        "id": "Jwd_klmR51N6"
      },
      "execution_count": 62,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Подготовка данных к машинному обучению"
      ],
      "metadata": {
        "id": "6guqvAjSIiKp"
      }
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "UUFQ10C0NB8u"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "from Bio import SeqIO\n",
        "from pandas.core import window\n",
        "input_file=\"chr1.fa\"\n",
        "fasta_sequence = SeqIO.read(open(input_file),'fasta')\n",
        "\n",
        "s=str(fasta_sequence.seq)\n",
        "window_size=100\n",
        "\n",
        "train_seq=[s[i:i+window_size] for i in  range(0,len(s)-window_size, window_size//2)]"
      ],
      "metadata": {
        "id": "RvGYg8EjKxp9"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "bed_start=[i for i in  range(0,len(s)-window_size, window_size//2)]\n",
        "bed_end=[i+window_size for i in  range(0,len(s)-window_size, window_size//2)]\n",
        "bed_chr=[\"chr1\" for i in  range(0,len(s)-window_size, window_size//2)]\n",
        "df=pd.DataFrame({\"chr1\":bed_chr,\"start\":bed_start,\"end\":bed_end})\n",
        "df.to_csv(\"tiles.bed\",header=False, index=False,sep=\"\\t\")"
      ],
      "metadata": {
        "id": "-9zkh1BJL4CD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools intersect -a tiles.bed -b target_chr1.bed -wao > dataset.tsv\n",
        "!bedtools map -a tiles.bed -b target_chr1.bed -c 5 -o mean > dataset2.tsv"
      ],
      "metadata": {
        "id": "Ewi-2K1xMAJX"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "overlaps=pd.read_csv(\"dataset2.tsv\",sep=\"\\t\",header=None)\n",
        "overlaps=overlaps.replace(\".\",0)\n",
        "dataset=pd.DataFrame({\"seq\":train_seq,\"score\":overlaps[3]})"
      ],
      "metadata": {
        "id": "Iluqh0B1MX4U"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Парсинг/обработка данных для обучения"
      ],
      "metadata": {
        "id": "3HYCRktUMx45"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def one_hot_encode(seq):\n",
        "    \n",
        "    seq=seq.upper() #<небольшое изменение\n",
        "    allowed = set(\"ACTGN\")\n",
        "    if not set(seq).issubset(allowed):\n",
        "        invalid = set(seq) - allowed\n",
        "        raise ValueError(f\"Sequence contains chars not in allowed DNA alphabet (ACGTN): {invalid}\")\n",
        "        \n",
        "    # Dictionary returning one-hot encoding for each nucleotide \n",
        "    nuc_d = {'A':[1.0,0.0,0.0,0.0],\n",
        "             'C':[0.0,1.0,0.0,0.0],\n",
        "             'G':[0.0,0.0,1.0,0.0],\n",
        "             'T':[0.0,0.0,0.0,1.0],\n",
        "             'N':[0.0,0.0,0.0,0.0]}\n",
        "    \n",
        "    # Create array from nucleotide sequence\n",
        "    vec=np.array([nuc_d[x] for x in seq])\n",
        "        \n",
        "    return vec\n",
        "\n",
        "\n",
        "def quick_split(df, split_frac=0.8, verbose=False):\n",
        "    '''\n",
        "    Given a df of samples, randomly split indices between\n",
        "    train and test at the desired fraction\n",
        "    '''\n",
        "    cols = df.columns # original columns, use to clean up reindexed cols\n",
        "    df = df.reset_index()\n",
        "\n",
        "    # shuffle indices\n",
        "    idxs = list(range(df.shape[0]))\n",
        "    random.shuffle(idxs)\n",
        "\n",
        "    # split shuffled index list by split_frac\n",
        "    split = int(len(idxs)*split_frac)\n",
        "    train_idxs = idxs[:split]\n",
        "    test_idxs = idxs[split:]\n",
        "    \n",
        "    # split dfs and return\n",
        "    train_df = df[df.index.isin(train_idxs)]\n",
        "    test_df = df[df.index.isin(test_idxs)]\n",
        "        \n",
        "    return train_df[cols], test_df[cols]"
      ],
      "metadata": {
        "id": "AOyd5UAUMpyT"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!wget "
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "x9pF95LZNB9v",
        "outputId": "448e0be7-1719-4ec7-928b-169a0ee50275"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-12-20 20:25:43--  https://drive.google.com/file/d/1c7UrHQ1owaPB2a_9CfWPU67ucORDphSa/view?usp=sharing\n",
            "Resolving drive.google.com (drive.google.com)... 142.251.2.102, 142.251.2.101, 142.251.2.113, ...\n",
            "Connecting to drive.google.com (drive.google.com)|142.251.2.102|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [text/html]\n",
            "Saving to: ‘view?usp=sharing’\n",
            "\n",
            "view?usp=sharing        [ <=>                ]  71.94K  --.-KB/s    in 0.004s  \n",
            "\n",
            "2022-12-20 20:25:43 (18.4 MB/s) - ‘view?usp=sharing’ saved [73665]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "full_train_df, test_df = quick_split(dataset)\n",
        "train_df, val_df = quick_split(full_train_df)\n",
        "\n",
        "print(\"Train:\", train_df.shape)\n",
        "print(\"Val:\", val_df.shape)\n",
        "print(\"Test:\", test_df.shape)\n",
        "\n",
        "train_df.head()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 258
        },
        "id": "HghCg65NNnej",
        "outputId": "ef1c500f-0176-4840-f90d-853597bca76d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Train: (3190406, 2)\n",
            "Val: (797602, 2)\n",
            "Test: (997003, 2)\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                 seq score\n",
              "1  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...     0\n",
              "2  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...     0\n",
              "3  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...     0\n",
              "4  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...     0\n",
              "5  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...     0"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-562c3f2b-64a1-462c-911d-6308c4adaf37\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>seq</th>\n",
              "      <th>score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-562c3f2b-64a1-462c-911d-6308c4adaf37')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-562c3f2b-64a1-462c-911d-6308c4adaf37 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-562c3f2b-64a1-462c-911d-6308c4adaf37');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    }
  ]
}
